{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9594f74",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\compat\\v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:From C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\util\\dispatch.py:1176: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.\n",
      "Epoch: 0, Step: 0, Loss: 7.90036\n",
      "Epoch: 0, Step: 10, Loss: 6.49546\n",
      "Epoch: 0, Step: 20, Loss: 6.76014\n",
      "Epoch: 0, Step: 30, Loss: 6.18253\n",
      "Epoch: 0, Step: 40, Loss: 6.18183\n",
      "Epoch: 0, Step: 50, Loss: 6.34937\n",
      "Epoch: 0, Step: 60, Loss: 6.00826\n",
      "Epoch: 0, Step: 70, Loss: 5.99322\n",
      "Epoch: 0, Step: 80, Loss: 6.1439\n",
      "Epoch: 0, Step: 90, Loss: 5.98492\n",
      "Epoch: 0, Step: 100, Loss: 5.94141\n",
      "Epoch: 0, Step: 110, Loss: 5.68805\n",
      "Epoch: 0, Step: 120, Loss: 5.71967\n",
      "Epoch: 0, Step: 130, Loss: 5.74442\n",
      "Epoch: 0, Step: 140, Loss: 5.62455\n",
      "Epoch: 0, Step: 150, Loss: 5.53852\n",
      "Epoch: 0, Step: 160, Loss: 5.95\n",
      "Epoch: 0, Step: 170, Loss: 5.47837\n",
      "Epoch: 0, Step: 180, Loss: 5.8573\n",
      "Epoch: 0, Step: 190, Loss: 5.42467\n",
      "Epoch: 0, Step: 200, Loss: 5.39844\n",
      "Epoch: 0, Step: 210, Loss: 5.20819\n",
      "Epoch: 0, Step: 220, Loss: 5.16771\n",
      "Epoch: 0, Step: 230, Loss: 5.29643\n",
      "Epoch: 0, Step: 240, Loss: 5.25439\n",
      "Epoch: 0, Step: 250, Loss: 5.23738\n",
      "Epoch: 0, Step: 260, Loss: 5.0196\n",
      "Epoch: 0, Step: 270, Loss: 4.95078\n",
      "Epoch: 0, Step: 280, Loss: 4.95643\n",
      "Epoch: 0, Step: 290, Loss: 4.94798\n",
      "Epoch: 0, Step: 300, Loss: 5.13739\n",
      "Epoch: 0, Step: 310, Loss: 4.94059\n",
      "Epoch: 0, Step: 320, Loss: 4.83505\n",
      "Epoch: 0, Step: 330, Loss: 4.59216\n",
      "Epoch: 0, Step: 340, Loss: 4.57897\n",
      "Epoch: 0, Step: 350, Loss: 4.68395\n",
      "Epoch: 0, Step: 360, Loss: 4.81054\n",
      "Epoch: 0, Step: 370, Loss: 4.48023\n",
      "Epoch: 0, Step: 380, Loss: 4.58135\n",
      "Epoch: 0, Step: 390, Loss: 4.6275\n",
      "Epoch: 0, Step: 400, Loss: 4.35707\n",
      "Epoch: 0, Step: 410, Loss: 4.4234\n",
      "Epoch: 0, Step: 420, Loss: 4.44967\n",
      "Epoch: 0, Step: 430, Loss: 4.27192\n",
      "Epoch: 0, Step: 440, Loss: 4.46713\n",
      "Epoch: 0, Step: 450, Loss: 4.19827\n",
      "Epoch: 1, Step: 100, Loss: 4.29627\n",
      "Epoch: 1, Step: 110, Loss: 4.3026\n",
      "Epoch: 1, Step: 120, Loss: 4.05636\n",
      "Epoch: 1, Step: 130, Loss: 4.02857\n",
      "Epoch: 1, Step: 140, Loss: 4.08781\n",
      "Epoch: 1, Step: 150, Loss: 3.98572\n",
      "Epoch: 1, Step: 160, Loss: 4.08252\n",
      "Epoch: 1, Step: 170, Loss: 3.951\n",
      "Epoch: 1, Step: 180, Loss: 4.1313\n",
      "Epoch: 1, Step: 190, Loss: 3.89161\n",
      "Epoch: 1, Step: 200, Loss: 4.14142\n",
      "Epoch: 1, Step: 210, Loss: 3.92738\n",
      "Epoch: 1, Step: 220, Loss: 4.09673\n",
      "Epoch: 1, Step: 230, Loss: 3.82471\n",
      "Epoch: 1, Step: 240, Loss: 3.65986\n",
      "Epoch: 1, Step: 250, Loss: 3.74562\n",
      "Epoch: 1, Step: 260, Loss: 3.72827\n",
      "Epoch: 1, Step: 270, Loss: 3.60155\n",
      "Epoch: 1, Step: 280, Loss: 3.90863\n",
      "Epoch: 1, Step: 290, Loss: 3.59777\n",
      "Epoch: 1, Step: 300, Loss: 3.51399\n",
      "Epoch: 1, Step: 310, Loss: 3.38709\n",
      "Epoch: 1, Step: 320, Loss: 3.57098\n",
      "Epoch: 1, Step: 330, Loss: 3.44147\n",
      "Epoch: 1, Step: 340, Loss: 3.46406\n",
      "Epoch: 1, Step: 350, Loss: 4.03505\n",
      "Epoch: 1, Step: 360, Loss: 3.48893\n",
      "Epoch: 1, Step: 370, Loss: 3.36684\n",
      "Epoch: 1, Step: 380, Loss: 3.36899\n",
      "Epoch: 1, Step: 390, Loss: 3.18075\n",
      "Epoch: 1, Step: 400, Loss: 3.35283\n",
      "Epoch: 1, Step: 410, Loss: 3.2996\n",
      "Epoch: 1, Step: 420, Loss: 3.38116\n",
      "Epoch: 1, Step: 430, Loss: 3.29882\n",
      "Epoch: 1, Step: 440, Loss: 3.2607\n",
      "Epoch: 1, Step: 450, Loss: 3.22228\n",
      "Epoch: 1, Step: 460, Loss: 3.13195\n",
      "Epoch: 1, Step: 470, Loss: 3.14198\n",
      "Epoch: 1, Step: 480, Loss: 3.15743\n",
      "Epoch: 1, Step: 490, Loss: 3.13796\n",
      "Epoch: 1, Step: 500, Loss: 3.2022\n",
      "Epoch: 1, Step: 510, Loss: 3.01049\n",
      "Epoch: 1, Step: 520, Loss: 3.24155\n",
      "Epoch: 1, Step: 530, Loss: 2.9587\n",
      "Epoch: 1, Step: 540, Loss: 3.11727\n",
      "Epoch: 1, Step: 550, Loss: 2.92475\n",
      "Epoch: 2, Step: 200, Loss: 3.09268\n",
      "Epoch: 2, Step: 210, Loss: 3.40457\n",
      "Epoch: 2, Step: 220, Loss: 2.95569\n",
      "Epoch: 2, Step: 230, Loss: 2.81409\n",
      "Epoch: 2, Step: 240, Loss: 3.00924\n",
      "Epoch: 2, Step: 250, Loss: 2.78446\n",
      "Epoch: 2, Step: 260, Loss: 2.78689\n",
      "Epoch: 2, Step: 270, Loss: 2.95105\n",
      "Epoch: 2, Step: 280, Loss: 2.78759\n",
      "Epoch: 2, Step: 290, Loss: 2.82917\n",
      "Epoch: 2, Step: 300, Loss: 2.73479\n",
      "Epoch: 2, Step: 310, Loss: 2.75142\n",
      "Epoch: 2, Step: 320, Loss: 2.67088\n",
      "Epoch: 2, Step: 330, Loss: 2.73197\n",
      "Epoch: 2, Step: 340, Loss: 2.69842\n",
      "Epoch: 2, Step: 350, Loss: 2.94739\n",
      "Epoch: 2, Step: 360, Loss: 2.63339\n",
      "Epoch: 2, Step: 370, Loss: 3.00467\n",
      "Epoch: 2, Step: 380, Loss: 2.68693\n",
      "Epoch: 2, Step: 390, Loss: 2.60124\n",
      "Epoch: 2, Step: 400, Loss: 2.56494\n",
      "Epoch: 2, Step: 410, Loss: 2.51532\n",
      "Epoch: 2, Step: 420, Loss: 2.63629\n",
      "Epoch: 2, Step: 430, Loss: 2.55522\n",
      "Epoch: 2, Step: 440, Loss: 2.53527\n",
      "Epoch: 2, Step: 450, Loss: 2.44027\n",
      "Epoch: 2, Step: 460, Loss: 2.46393\n",
      "Epoch: 2, Step: 470, Loss: 2.46834\n",
      "Epoch: 2, Step: 480, Loss: 2.52089\n",
      "Epoch: 2, Step: 490, Loss: 2.63577\n",
      "Epoch: 2, Step: 500, Loss: 2.40804\n",
      "Epoch: 2, Step: 510, Loss: 2.46533\n",
      "Epoch: 2, Step: 520, Loss: 2.29408\n",
      "Epoch: 2, Step: 530, Loss: 2.28489\n",
      "Epoch: 2, Step: 540, Loss: 2.34929\n",
      "Epoch: 2, Step: 550, Loss: 2.57807\n",
      "Epoch: 2, Step: 560, Loss: 2.2502\n",
      "Epoch: 2, Step: 570, Loss: 2.32372\n",
      "Epoch: 2, Step: 580, Loss: 2.45029\n",
      "Epoch: 2, Step: 590, Loss: 2.26595\n",
      "Epoch: 2, Step: 600, Loss: 2.32533\n",
      "Epoch: 2, Step: 610, Loss: 2.35723\n",
      "Epoch: 2, Step: 620, Loss: 2.18606\n",
      "Epoch: 2, Step: 630, Loss: 2.44009\n",
      "Epoch: 2, Step: 640, Loss: 2.21026\n",
      "Epoch: 2, Step: 650, Loss: 2.2387\n",
      "Epoch: 3, Step: 300, Loss: 2.30654\n",
      "Epoch: 3, Step: 310, Loss: 2.15229\n",
      "Epoch: 3, Step: 320, Loss: 2.13092\n",
      "Epoch: 3, Step: 330, Loss: 2.22478\n",
      "Epoch: 3, Step: 340, Loss: 2.23698\n",
      "Epoch: 3, Step: 350, Loss: 2.12116\n",
      "Epoch: 3, Step: 360, Loss: 2.16505\n",
      "Epoch: 3, Step: 370, Loss: 2.31292\n",
      "Epoch: 3, Step: 380, Loss: 2.12902\n",
      "Epoch: 3, Step: 390, Loss: 2.2017\n",
      "Epoch: 3, Step: 400, Loss: 2.14177\n",
      "Epoch: 3, Step: 410, Loss: 2.18527\n",
      "Epoch: 3, Step: 420, Loss: 2.08259\n",
      "Epoch: 3, Step: 430, Loss: 2.02277\n",
      "Epoch: 3, Step: 440, Loss: 2.03701\n",
      "Epoch: 3, Step: 450, Loss: 2.09885\n",
      "Epoch: 3, Step: 460, Loss: 1.97396\n",
      "Epoch: 3, Step: 470, Loss: 2.14847\n",
      "Epoch: 3, Step: 480, Loss: 1.98041\n",
      "Epoch: 3, Step: 490, Loss: 1.97463\n",
      "Epoch: 3, Step: 500, Loss: 1.93244\n",
      "Epoch: 3, Step: 510, Loss: 1.98326\n",
      "Epoch: 3, Step: 520, Loss: 1.92405\n",
      "Epoch: 3, Step: 530, Loss: 2.32608\n",
      "Epoch: 3, Step: 540, Loss: 2.12523\n",
      "Epoch: 3, Step: 550, Loss: 1.99862\n",
      "Epoch: 3, Step: 560, Loss: 1.89508\n",
      "Epoch: 3, Step: 570, Loss: 2.01256\n",
      "Epoch: 3, Step: 580, Loss: 1.84914\n",
      "Epoch: 3, Step: 590, Loss: 1.90614\n",
      "Epoch: 3, Step: 600, Loss: 1.88012\n",
      "Epoch: 3, Step: 610, Loss: 2.01728\n",
      "Epoch: 3, Step: 620, Loss: 1.87683\n",
      "Epoch: 3, Step: 630, Loss: 1.90017\n",
      "Epoch: 3, Step: 640, Loss: 1.80266\n",
      "Epoch: 3, Step: 650, Loss: 1.80199\n",
      "Epoch: 3, Step: 660, Loss: 1.81563\n",
      "Epoch: 3, Step: 670, Loss: 1.81337\n",
      "Epoch: 3, Step: 680, Loss: 1.90312\n",
      "Epoch: 3, Step: 690, Loss: 1.89074\n",
      "Epoch: 3, Step: 700, Loss: 1.76606\n",
      "Epoch: 3, Step: 710, Loss: 1.90159\n",
      "Epoch: 3, Step: 720, Loss: 1.79418\n",
      "Epoch: 3, Step: 730, Loss: 1.80383\n",
      "Epoch: 3, Step: 740, Loss: 1.73944\n",
      "Epoch: 3, Step: 750, Loss: 1.90931\n",
      "Epoch: 4, Step: 400, Loss: 2.05441\n",
      "Epoch: 4, Step: 410, Loss: 1.75715\n",
      "Epoch: 4, Step: 420, Loss: 1.76799\n",
      "Epoch: 4, Step: 430, Loss: 1.74105\n",
      "Epoch: 4, Step: 440, Loss: 1.71846\n",
      "Epoch: 4, Step: 450, Loss: 1.72586\n",
      "Epoch: 4, Step: 460, Loss: 1.7222\n",
      "Epoch: 4, Step: 470, Loss: 1.7113\n",
      "Epoch: 4, Step: 480, Loss: 1.72031\n",
      "Epoch: 4, Step: 490, Loss: 1.66194\n",
      "Epoch: 4, Step: 500, Loss: 1.6874\n",
      "Epoch: 4, Step: 510, Loss: 1.66167\n",
      "Epoch: 4, Step: 520, Loss: 1.68369\n",
      "Epoch: 4, Step: 530, Loss: 1.77014\n",
      "Epoch: 4, Step: 540, Loss: 1.75117\n",
      "Epoch: 4, Step: 550, Loss: 1.75575\n",
      "Epoch: 4, Step: 560, Loss: 1.85563\n",
      "Epoch: 4, Step: 570, Loss: 1.62816\n",
      "Epoch: 4, Step: 580, Loss: 1.62943\n",
      "Epoch: 4, Step: 590, Loss: 1.5933\n",
      "Epoch: 4, Step: 600, Loss: 1.60708\n",
      "Epoch: 4, Step: 610, Loss: 1.60862\n",
      "Epoch: 4, Step: 620, Loss: 1.61915\n",
      "Epoch: 4, Step: 630, Loss: 1.58724\n",
      "Epoch: 4, Step: 640, Loss: 1.5384\n",
      "Epoch: 4, Step: 650, Loss: 1.58558\n",
      "Epoch: 4, Step: 660, Loss: 1.53234\n",
      "Epoch: 4, Step: 670, Loss: 1.61011\n",
      "Epoch: 4, Step: 680, Loss: 1.61279\n",
      "Epoch: 4, Step: 690, Loss: 1.55844\n",
      "Epoch: 4, Step: 700, Loss: 1.52032\n",
      "Epoch: 4, Step: 710, Loss: 1.5106\n",
      "Epoch: 4, Step: 720, Loss: 1.49154\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch: 4, Step: 730, Loss: 1.50163\n",
      "Epoch: 4, Step: 740, Loss: 1.61748\n",
      "Epoch: 4, Step: 750, Loss: 1.47772\n",
      "Epoch: 4, Step: 760, Loss: 1.48187\n",
      "Epoch: 4, Step: 770, Loss: 1.61709\n",
      "Epoch: 4, Step: 780, Loss: 1.45378\n",
      "Epoch: 4, Step: 790, Loss: 1.59031\n",
      "Epoch: 4, Step: 800, Loss: 1.47845\n",
      "Epoch: 4, Step: 810, Loss: 1.4773\n",
      "Epoch: 4, Step: 820, Loss: 1.46995\n",
      "Epoch: 4, Step: 830, Loss: 1.50312\n",
      "Epoch: 4, Step: 840, Loss: 1.44461\n",
      "Epoch: 4, Step: 850, Loss: 1.45808\n",
      "Epoch: 5, Step: 500, Loss: 1.41608\n",
      "Epoch: 5, Step: 510, Loss: 1.40913\n",
      "Epoch: 5, Step: 520, Loss: 1.42705\n",
      "Epoch: 5, Step: 530, Loss: 1.49648\n",
      "Epoch: 5, Step: 540, Loss: 1.40196\n",
      "Epoch: 5, Step: 550, Loss: 1.49939\n",
      "Epoch: 5, Step: 560, Loss: 1.43256\n",
      "Epoch: 5, Step: 570, Loss: 1.38923\n",
      "Epoch: 5, Step: 580, Loss: 1.42233\n",
      "Epoch: 5, Step: 590, Loss: 1.42442\n",
      "Epoch: 5, Step: 600, Loss: 1.45062\n",
      "Epoch: 5, Step: 610, Loss: 1.37909\n",
      "Epoch: 5, Step: 620, Loss: 1.34393\n",
      "Epoch: 5, Step: 630, Loss: 1.35616\n",
      "Epoch: 5, Step: 640, Loss: 1.39213\n",
      "Epoch: 5, Step: 650, Loss: 1.31046\n",
      "Epoch: 5, Step: 660, Loss: 1.42006\n",
      "Epoch: 5, Step: 670, Loss: 1.36779\n",
      "Epoch: 5, Step: 680, Loss: 1.31962\n",
      "Epoch: 5, Step: 690, Loss: 1.29061\n",
      "Epoch: 5, Step: 700, Loss: 1.31862\n",
      "Epoch: 5, Step: 710, Loss: 1.31102\n",
      "Epoch: 5, Step: 720, Loss: 1.63205\n",
      "Epoch: 5, Step: 730, Loss: 1.44352\n",
      "Epoch: 5, Step: 740, Loss: 1.29069\n",
      "Epoch: 5, Step: 750, Loss: 1.28733\n",
      "Epoch: 5, Step: 760, Loss: 1.26681\n",
      "Epoch: 5, Step: 770, Loss: 1.2723\n",
      "Epoch: 5, Step: 780, Loss: 1.26121\n",
      "Epoch: 5, Step: 790, Loss: 1.25835\n",
      "Epoch: 5, Step: 800, Loss: 1.31255\n",
      "Epoch: 5, Step: 810, Loss: 1.2485\n",
      "Epoch: 5, Step: 820, Loss: 1.28075\n",
      "Epoch: 5, Step: 830, Loss: 1.22547\n",
      "Epoch: 5, Step: 840, Loss: 1.22689\n",
      "Epoch: 5, Step: 850, Loss: 1.2195\n",
      "Epoch: 5, Step: 860, Loss: 1.23546\n",
      "Epoch: 5, Step: 870, Loss: 1.3263\n",
      "Epoch: 5, Step: 880, Loss: 1.21748\n",
      "Epoch: 5, Step: 890, Loss: 1.19988\n",
      "Epoch: 5, Step: 900, Loss: 1.25485\n",
      "Epoch: 5, Step: 910, Loss: 1.23967\n",
      "Epoch: 5, Step: 920, Loss: 1.19087\n",
      "Epoch: 5, Step: 930, Loss: 1.19027\n",
      "Epoch: 5, Step: 940, Loss: 1.59197\n",
      "Epoch: 5, Step: 950, Loss: 1.1742\n",
      "Epoch: 6, Step: 600, Loss: 1.21198\n",
      "Epoch: 6, Step: 610, Loss: 1.22134\n",
      "Epoch: 6, Step: 620, Loss: 1.18658\n",
      "Epoch: 6, Step: 630, Loss: 1.15534\n",
      "Epoch: 6, Step: 640, Loss: 1.18384\n",
      "Epoch: 6, Step: 650, Loss: 1.17608\n",
      "Epoch: 6, Step: 660, Loss: 1.15247\n",
      "Epoch: 6, Step: 670, Loss: 1.20665\n",
      "Epoch: 6, Step: 680, Loss: 1.14341\n",
      "Epoch: 6, Step: 690, Loss: 1.15302\n",
      "Epoch: 6, Step: 700, Loss: 1.17507\n",
      "Epoch: 6, Step: 710, Loss: 1.13239\n",
      "Epoch: 6, Step: 720, Loss: 1.32948\n",
      "Epoch: 6, Step: 730, Loss: 1.13292\n",
      "Epoch: 6, Step: 740, Loss: 1.41017\n",
      "Epoch: 6, Step: 750, Loss: 1.1362\n",
      "Epoch: 6, Step: 760, Loss: 1.12176\n",
      "Epoch: 6, Step: 770, Loss: 1.15646\n",
      "Epoch: 6, Step: 780, Loss: 1.11066\n",
      "Epoch: 6, Step: 790, Loss: 1.10843\n",
      "Epoch: 6, Step: 800, Loss: 1.13293\n",
      "Epoch: 6, Step: 810, Loss: 1.08986\n",
      "Epoch: 6, Step: 820, Loss: 1.09003\n",
      "Epoch: 6, Step: 830, Loss: 1.0686\n",
      "Epoch: 6, Step: 840, Loss: 1.08318\n",
      "Epoch: 6, Step: 850, Loss: 1.09496\n",
      "Epoch: 6, Step: 860, Loss: 1.10969\n",
      "Epoch: 6, Step: 870, Loss: 1.09882\n",
      "Epoch: 6, Step: 880, Loss: 1.05365\n",
      "Epoch: 6, Step: 890, Loss: 1.04897\n",
      "Epoch: 6, Step: 900, Loss: 1.03668\n",
      "Epoch: 6, Step: 910, Loss: 1.04502\n",
      "Epoch: 6, Step: 920, Loss: 1.04829\n",
      "Epoch: 6, Step: 930, Loss: 1.09167\n",
      "Epoch: 6, Step: 940, Loss: 1.04006\n",
      "Epoch: 6, Step: 950, Loss: 1.01748\n",
      "Epoch: 6, Step: 960, Loss: 1.12777\n",
      "Epoch: 6, Step: 970, Loss: 1.00613\n",
      "Epoch: 6, Step: 980, Loss: 1.09066\n",
      "Epoch: 6, Step: 990, Loss: 1.02359\n",
      "Epoch: 6, Step: 1000, Loss: 1.01698\n",
      "Epoch: 6, Step: 1010, Loss: 1.02322\n",
      "Epoch: 6, Step: 1020, Loss: 1.04942\n",
      "Epoch: 6, Step: 1030, Loss: 1.00172\n",
      "Epoch: 6, Step: 1040, Loss: 1.02184\n",
      "Epoch: 6, Step: 1050, Loss: 0.989161\n",
      "Epoch: 7, Step: 700, Loss: 0.987224\n",
      "Epoch: 7, Step: 710, Loss: 0.993705\n",
      "Epoch: 7, Step: 720, Loss: 1.01336\n",
      "Epoch: 7, Step: 730, Loss: 1.01216\n",
      "Epoch: 7, Step: 740, Loss: 1.00455\n",
      "Epoch: 7, Step: 750, Loss: 0.985562\n",
      "Epoch: 7, Step: 760, Loss: 0.966959\n",
      "Epoch: 7, Step: 770, Loss: 1.02443\n",
      "Epoch: 7, Step: 780, Loss: 0.984683\n",
      "Epoch: 7, Step: 790, Loss: 0.994273\n",
      "Epoch: 7, Step: 800, Loss: 0.940619\n",
      "Epoch: 7, Step: 810, Loss: 0.945302\n",
      "Epoch: 7, Step: 820, Loss: 0.957858\n",
      "Epoch: 7, Step: 830, Loss: 0.94648\n",
      "Epoch: 7, Step: 840, Loss: 0.923698\n",
      "Epoch: 7, Step: 850, Loss: 0.99746\n",
      "Epoch: 7, Step: 860, Loss: 0.914939\n",
      "Epoch: 7, Step: 870, Loss: 0.912647\n",
      "Epoch: 7, Step: 880, Loss: 0.933767\n",
      "Epoch: 7, Step: 890, Loss: 0.943842\n",
      "Epoch: 7, Step: 900, Loss: 0.910508\n",
      "Epoch: 7, Step: 910, Loss: 1.20913\n",
      "Epoch: 7, Step: 920, Loss: 0.952049\n",
      "Epoch: 7, Step: 930, Loss: 0.902145\n",
      "Epoch: 7, Step: 940, Loss: 0.907048\n",
      "Epoch: 7, Step: 950, Loss: 0.897509\n",
      "Epoch: 7, Step: 960, Loss: 0.885966\n",
      "Epoch: 7, Step: 970, Loss: 0.894982\n",
      "Epoch: 7, Step: 980, Loss: 0.924707\n",
      "Epoch: 7, Step: 990, Loss: 0.905961\n",
      "Epoch: 7, Step: 1000, Loss: 0.895555\n",
      "Epoch: 7, Step: 1010, Loss: 0.894632\n",
      "Epoch: 7, Step: 1020, Loss: 0.861468\n",
      "Epoch: 7, Step: 1030, Loss: 0.8608\n",
      "Epoch: 7, Step: 1040, Loss: 0.866751\n",
      "Epoch: 7, Step: 1050, Loss: 0.872933\n",
      "Epoch: 7, Step: 1060, Loss: 0.875653\n",
      "Epoch: 7, Step: 1070, Loss: 0.854901\n",
      "Epoch: 7, Step: 1080, Loss: 0.866737\n",
      "Epoch: 7, Step: 1090, Loss: 0.854752\n",
      "Epoch: 7, Step: 1100, Loss: 0.900211\n",
      "Epoch: 7, Step: 1110, Loss: 0.852216\n",
      "Epoch: 7, Step: 1120, Loss: 0.926096\n",
      "Epoch: 7, Step: 1130, Loss: 1.10448\n",
      "Epoch: 7, Step: 1140, Loss: 0.830272\n",
      "Epoch: 7, Step: 1150, Loss: 0.853459\n",
      "Epoch: 8, Step: 800, Loss: 0.85621\n",
      "Epoch: 8, Step: 810, Loss: 0.823628\n",
      "Epoch: 8, Step: 820, Loss: 0.816007\n",
      "Epoch: 8, Step: 830, Loss: 0.842987\n",
      "Epoch: 8, Step: 840, Loss: 0.829532\n",
      "Epoch: 8, Step: 850, Loss: 0.826931\n",
      "Epoch: 8, Step: 860, Loss: 0.850059\n",
      "Epoch: 8, Step: 870, Loss: 0.795135\n",
      "Epoch: 8, Step: 880, Loss: 0.830811\n",
      "Epoch: 8, Step: 890, Loss: 0.817995\n",
      "Epoch: 8, Step: 900, Loss: 0.801715\n",
      "Epoch: 8, Step: 910, Loss: 0.980942\n",
      "Epoch: 8, Step: 920, Loss: 0.783782\n",
      "Epoch: 8, Step: 930, Loss: 1.02666\n",
      "Epoch: 8, Step: 940, Loss: 0.822837\n",
      "Epoch: 8, Step: 950, Loss: 0.778184\n",
      "Epoch: 8, Step: 960, Loss: 0.831576\n",
      "Epoch: 8, Step: 970, Loss: 0.788852\n",
      "Epoch: 8, Step: 980, Loss: 0.781273\n",
      "Epoch: 8, Step: 990, Loss: 0.785269\n",
      "Epoch: 8, Step: 1000, Loss: 0.778114\n",
      "Epoch: 8, Step: 1010, Loss: 0.760954\n",
      "Epoch: 8, Step: 1020, Loss: 0.75453\n",
      "Epoch: 8, Step: 1030, Loss: 0.776449\n",
      "Epoch: 8, Step: 1040, Loss: 0.766731\n",
      "Epoch: 8, Step: 1050, Loss: 0.765657\n",
      "Epoch: 8, Step: 1060, Loss: 0.785922\n",
      "Epoch: 8, Step: 1070, Loss: 0.744708\n",
      "Epoch: 8, Step: 1080, Loss: 0.754117\n",
      "Epoch: 8, Step: 1090, Loss: 0.735487\n",
      "Epoch: 8, Step: 1100, Loss: 0.757099\n",
      "Epoch: 8, Step: 1110, Loss: 0.739746\n",
      "Epoch: 8, Step: 1120, Loss: 0.763366\n",
      "Epoch: 8, Step: 1130, Loss: 0.731552\n",
      "Epoch: 8, Step: 1140, Loss: 0.72486\n",
      "Epoch: 8, Step: 1150, Loss: 0.798111\n",
      "Epoch: 8, Step: 1160, Loss: 0.719725\n",
      "Epoch: 8, Step: 1170, Loss: 0.771578\n",
      "Epoch: 8, Step: 1180, Loss: 0.732612\n",
      "Epoch: 8, Step: 1190, Loss: 0.729962\n",
      "Epoch: 8, Step: 1200, Loss: 0.726322\n",
      "Epoch: 8, Step: 1210, Loss: 0.743623\n",
      "Epoch: 8, Step: 1220, Loss: 0.731683\n",
      "Epoch: 8, Step: 1230, Loss: 0.706081\n",
      "Epoch: 8, Step: 1240, Loss: 0.694634\n",
      "Epoch: 8, Step: 1250, Loss: 0.700748\n",
      "Epoch: 9, Step: 900, Loss: 0.695024\n",
      "Epoch: 9, Step: 910, Loss: 0.747734\n",
      "Epoch: 9, Step: 920, Loss: 0.690996\n",
      "Epoch: 9, Step: 930, Loss: 0.783347\n",
      "Epoch: 9, Step: 940, Loss: 0.685379\n",
      "Epoch: 9, Step: 950, Loss: 0.690133\n",
      "Epoch: 9, Step: 960, Loss: 0.72809\n",
      "Epoch: 9, Step: 970, Loss: 0.701118\n",
      "Epoch: 9, Step: 980, Loss: 0.696234\n",
      "Epoch: 9, Step: 990, Loss: 0.671859\n",
      "Epoch: 9, Step: 1000, Loss: 0.665613\n",
      "Epoch: 9, Step: 1010, Loss: 0.681442\n",
      "Epoch: 9, Step: 1020, Loss: 0.671595\n",
      "Epoch: 9, Step: 1030, Loss: 0.654962\n",
      "Epoch: 9, Step: 1040, Loss: 0.72907\n",
      "Epoch: 9, Step: 1050, Loss: 0.668916\n",
      "Epoch: 9, Step: 1060, Loss: 0.646156\n",
      "Epoch: 9, Step: 1070, Loss: 0.6458\n",
      "Epoch: 9, Step: 1080, Loss: 0.653117\n",
      "Epoch: 9, Step: 1090, Loss: 0.661171\n",
      "Epoch: 9, Step: 1100, Loss: 0.880442\n",
      "Epoch: 9, Step: 1110, Loss: 0.650551\n",
      "Epoch: 9, Step: 1120, Loss: 0.643054\n",
      "Epoch: 9, Step: 1130, Loss: 0.655937\n",
      "Epoch: 9, Step: 1140, Loss: 0.639411\n",
      "Epoch: 9, Step: 1150, Loss: 0.634548\n",
      "Epoch: 9, Step: 1160, Loss: 0.630358\n",
      "Epoch: 9, Step: 1170, Loss: 0.648268\n",
      "Epoch: 9, Step: 1180, Loss: 0.654567\n",
      "Epoch: 9, Step: 1190, Loss: 0.639399\n",
      "Epoch: 9, Step: 1200, Loss: 0.640552\n",
      "Epoch: 9, Step: 1210, Loss: 0.611711\n",
      "Epoch: 9, Step: 1220, Loss: 0.614174\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch: 9, Step: 1230, Loss: 0.623571\n",
      "Epoch: 9, Step: 1240, Loss: 0.608197\n",
      "Epoch: 9, Step: 1250, Loss: 0.62526\n",
      "Epoch: 9, Step: 1260, Loss: 0.617914\n",
      "Epoch: 9, Step: 1270, Loss: 0.628673\n",
      "Epoch: 9, Step: 1280, Loss: 0.610496\n",
      "Epoch: 9, Step: 1290, Loss: 0.695551\n",
      "Epoch: 9, Step: 1300, Loss: 0.602265\n",
      "Epoch: 9, Step: 1310, Loss: 0.642538\n",
      "Epoch: 9, Step: 1320, Loss: 0.86381\n",
      "Epoch: 9, Step: 1330, Loss: 0.588438\n",
      "Epoch: 9, Step: 1340, Loss: 0.599753\n",
      "Epoch: 9, Step: 1350, Loss: 0.62418\n",
      "Epoch: 10, Step: 1000, Loss: 0.596529\n",
      "Epoch: 10, Step: 1010, Loss: 0.587021\n",
      "Epoch: 10, Step: 1020, Loss: 0.602959\n",
      "Epoch: 10, Step: 1030, Loss: 0.591999\n",
      "Epoch: 10, Step: 1040, Loss: 0.637011\n",
      "Epoch: 10, Step: 1050, Loss: 0.57598\n",
      "Epoch: 10, Step: 1060, Loss: 0.570022\n",
      "Epoch: 10, Step: 1070, Loss: 0.587254\n",
      "Epoch: 10, Step: 1080, Loss: 0.595894\n",
      "Epoch: 10, Step: 1090, Loss: 0.571204\n",
      "Epoch: 10, Step: 1100, Loss: 0.715298\n",
      "Epoch: 10, Step: 1110, Loss: 0.576132\n",
      "Epoch: 10, Step: 1120, Loss: 0.762525\n",
      "Epoch: 10, Step: 1130, Loss: 0.595266\n",
      "Epoch: 10, Step: 1140, Loss: 0.633599\n",
      "Epoch: 10, Step: 1150, Loss: 0.562284\n",
      "Epoch: 10, Step: 1160, Loss: 0.575673\n",
      "Epoch: 10, Step: 1170, Loss: 0.560485\n",
      "Epoch: 10, Step: 1180, Loss: 0.566016\n",
      "Epoch: 10, Step: 1190, Loss: 0.560084\n",
      "Epoch: 10, Step: 1200, Loss: 0.550137\n",
      "Epoch: 10, Step: 1210, Loss: 0.545341\n",
      "Epoch: 10, Step: 1220, Loss: 0.555442\n",
      "Epoch: 10, Step: 1230, Loss: 0.554916\n",
      "Epoch: 10, Step: 1240, Loss: 0.565453\n",
      "Epoch: 10, Step: 1250, Loss: 0.56239\n",
      "Epoch: 10, Step: 1260, Loss: 0.54542\n",
      "Epoch: 10, Step: 1270, Loss: 0.531964\n",
      "Epoch: 10, Step: 1280, Loss: 0.536016\n",
      "Epoch: 10, Step: 1290, Loss: 0.556239\n",
      "Epoch: 10, Step: 1300, Loss: 0.569723\n",
      "Epoch: 10, Step: 1310, Loss: 0.523658\n",
      "Epoch: 10, Step: 1320, Loss: 0.53566\n",
      "Epoch: 10, Step: 1330, Loss: 0.633465\n",
      "Epoch: 10, Step: 1340, Loss: 0.52231\n",
      "Epoch: 10, Step: 1350, Loss: 0.535121\n",
      "Epoch: 10, Step: 1360, Loss: 0.616944\n",
      "Epoch: 10, Step: 1370, Loss: 0.52158\n",
      "Epoch: 10, Step: 1380, Loss: 0.534036\n",
      "Epoch: 10, Step: 1390, Loss: 0.512441\n",
      "Epoch: 10, Step: 1400, Loss: 0.550919\n",
      "Epoch: 10, Step: 1410, Loss: 0.537226\n",
      "Epoch: 10, Step: 1420, Loss: 0.514178\n",
      "Epoch: 10, Step: 1430, Loss: 0.514682\n",
      "Epoch: 10, Step: 1440, Loss: 0.51685\n",
      "Epoch: 10, Step: 1450, Loss: 0.530031\n",
      "Epoch: 11, Step: 1100, Loss: 0.677287\n",
      "Epoch: 11, Step: 1110, Loss: 0.524838\n",
      "Epoch: 11, Step: 1120, Loss: 0.616399\n",
      "Epoch: 11, Step: 1130, Loss: 0.503393\n",
      "Epoch: 11, Step: 1140, Loss: 0.539106\n",
      "Epoch: 11, Step: 1150, Loss: 0.536279\n",
      "Epoch: 11, Step: 1160, Loss: 0.524326\n",
      "Epoch: 11, Step: 1170, Loss: 0.503434\n",
      "Epoch: 11, Step: 1180, Loss: 0.496288\n",
      "Epoch: 11, Step: 1190, Loss: 0.495616\n",
      "Epoch: 11, Step: 1200, Loss: 0.535173\n",
      "Epoch: 11, Step: 1210, Loss: 0.523713\n",
      "Epoch: 11, Step: 1220, Loss: 0.550934\n",
      "Epoch: 11, Step: 1230, Loss: 0.489357\n",
      "Epoch: 11, Step: 1240, Loss: 0.496498\n",
      "Epoch: 11, Step: 1250, Loss: 0.478186\n",
      "Epoch: 11, Step: 1260, Loss: 0.480121\n",
      "Epoch: 11, Step: 1270, Loss: 0.491682\n",
      "Epoch: 11, Step: 1280, Loss: 0.482318\n",
      "Epoch: 11, Step: 1290, Loss: 0.710451\n",
      "Epoch: 11, Step: 1300, Loss: 0.492226\n",
      "Epoch: 11, Step: 1310, Loss: 0.496801\n",
      "Epoch: 11, Step: 1320, Loss: 0.480916\n",
      "Epoch: 11, Step: 1330, Loss: 0.472174\n",
      "Epoch: 11, Step: 1340, Loss: 0.469737\n",
      "Epoch: 11, Step: 1350, Loss: 0.471969\n",
      "Epoch: 11, Step: 1360, Loss: 0.491687\n",
      "Epoch: 11, Step: 1370, Loss: 0.493493\n",
      "Epoch: 11, Step: 1380, Loss: 0.481678\n",
      "Epoch: 11, Step: 1390, Loss: 0.480537\n",
      "Epoch: 11, Step: 1400, Loss: 0.457645\n",
      "Epoch: 11, Step: 1410, Loss: 0.458475\n",
      "Epoch: 11, Step: 1420, Loss: 0.476793\n",
      "Epoch: 11, Step: 1430, Loss: 0.461422\n",
      "Epoch: 11, Step: 1440, Loss: 0.477968\n",
      "Epoch: 11, Step: 1450, Loss: 0.453148\n",
      "Epoch: 11, Step: 1460, Loss: 0.471889\n",
      "Epoch: 11, Step: 1470, Loss: 0.462411\n",
      "Epoch: 11, Step: 1480, Loss: 0.526196\n",
      "Epoch: 11, Step: 1490, Loss: 0.450974\n",
      "Epoch: 11, Step: 1500, Loss: 0.483302\n",
      "Epoch: 11, Step: 1510, Loss: 0.708627\n",
      "Epoch: 11, Step: 1520, Loss: 0.453993\n",
      "Epoch: 11, Step: 1530, Loss: 0.441266\n",
      "Epoch: 11, Step: 1540, Loss: 0.469506\n",
      "Epoch: 11, Step: 1550, Loss: 0.451706\n",
      "Epoch: 12, Step: 1200, Loss: 0.449542\n",
      "Epoch: 12, Step: 1210, Loss: 0.462747\n",
      "Epoch: 12, Step: 1220, Loss: 0.44195\n",
      "Epoch: 12, Step: 1230, Loss: 0.486704\n",
      "Epoch: 12, Step: 1240, Loss: 0.43479\n",
      "Epoch: 12, Step: 1250, Loss: 0.434851\n",
      "Epoch: 12, Step: 1260, Loss: 0.435395\n",
      "Epoch: 12, Step: 1270, Loss: 0.447722\n",
      "Epoch: 12, Step: 1280, Loss: 0.439692\n",
      "Epoch: 12, Step: 1290, Loss: 0.610858\n",
      "Epoch: 12, Step: 1300, Loss: 0.431401\n",
      "Epoch: 12, Step: 1310, Loss: 0.59578\n",
      "Epoch: 12, Step: 1320, Loss: 0.438341\n",
      "Epoch: 12, Step: 1330, Loss: 0.490047\n",
      "Epoch: 12, Step: 1340, Loss: 0.41943\n",
      "Epoch: 12, Step: 1350, Loss: 0.432373\n",
      "Epoch: 12, Step: 1360, Loss: 0.436014\n",
      "Epoch: 12, Step: 1370, Loss: 0.419989\n",
      "Epoch: 12, Step: 1380, Loss: 0.515024\n",
      "Epoch: 12, Step: 1390, Loss: 0.418184\n",
      "Epoch: 12, Step: 1400, Loss: 0.424447\n",
      "Epoch: 12, Step: 1410, Loss: 0.431114\n",
      "Epoch: 12, Step: 1420, Loss: 0.436325\n",
      "Epoch: 12, Step: 1430, Loss: 0.420928\n",
      "Epoch: 12, Step: 1440, Loss: 0.439246\n",
      "Epoch: 12, Step: 1450, Loss: 0.423111\n",
      "Epoch: 12, Step: 1460, Loss: 0.454322\n",
      "Epoch: 12, Step: 1470, Loss: 0.406566\n",
      "Epoch: 12, Step: 1480, Loss: 0.422989\n",
      "Epoch: 12, Step: 1490, Loss: 0.43731\n",
      "Epoch: 12, Step: 1500, Loss: 0.403371\n",
      "Epoch: 12, Step: 1510, Loss: 0.404991\n",
      "Epoch: 12, Step: 1520, Loss: 0.446039\n",
      "Epoch: 12, Step: 1530, Loss: 0.397963\n",
      "Epoch: 12, Step: 1540, Loss: 0.435323\n",
      "Epoch: 12, Step: 1550, Loss: 0.439502\n",
      "Epoch: 12, Step: 1560, Loss: 0.393758\n",
      "Epoch: 12, Step: 1570, Loss: 0.408782\n",
      "Epoch: 12, Step: 1580, Loss: 0.40987\n",
      "Epoch: 12, Step: 1590, Loss: 0.398944\n",
      "Epoch: 12, Step: 1600, Loss: 0.411293\n",
      "Epoch: 12, Step: 1610, Loss: 0.387813\n",
      "Epoch: 12, Step: 1620, Loss: 0.387941\n",
      "Epoch: 12, Step: 1630, Loss: 0.388869\n",
      "Epoch: 12, Step: 1640, Loss: 0.386564\n",
      "Epoch: 12, Step: 1650, Loss: 0.414427\n",
      "Epoch: 13, Step: 1300, Loss: 0.387538\n",
      "Epoch: 13, Step: 1310, Loss: 0.489003\n",
      "Epoch: 13, Step: 1320, Loss: 0.378242\n",
      "Epoch: 13, Step: 1330, Loss: 0.415256\n",
      "Epoch: 13, Step: 1340, Loss: 0.407898\n",
      "Epoch: 13, Step: 1350, Loss: 0.405796\n",
      "Epoch: 13, Step: 1360, Loss: 0.379407\n",
      "Epoch: 13, Step: 1370, Loss: 0.376607\n",
      "Epoch: 13, Step: 1380, Loss: 0.388232\n",
      "Epoch: 13, Step: 1390, Loss: 0.383002\n",
      "Epoch: 13, Step: 1400, Loss: 0.384291\n",
      "Epoch: 13, Step: 1410, Loss: 0.431961\n",
      "Epoch: 13, Step: 1420, Loss: 0.378113\n",
      "Epoch: 13, Step: 1430, Loss: 0.382513\n",
      "Epoch: 13, Step: 1440, Loss: 0.367588\n",
      "Epoch: 13, Step: 1450, Loss: 0.367154\n",
      "Epoch: 13, Step: 1460, Loss: 0.383417\n",
      "Epoch: 13, Step: 1470, Loss: 0.582394\n",
      "Epoch: 13, Step: 1480, Loss: 0.380063\n",
      "Epoch: 13, Step: 1490, Loss: 0.368746\n",
      "Epoch: 13, Step: 1500, Loss: 0.402335\n",
      "Epoch: 13, Step: 1510, Loss: 0.36916\n",
      "Epoch: 13, Step: 1520, Loss: 0.371876\n",
      "Epoch: 13, Step: 1530, Loss: 0.371963\n",
      "Epoch: 13, Step: 1540, Loss: 0.380542\n",
      "Epoch: 13, Step: 1550, Loss: 0.409539\n",
      "Epoch: 13, Step: 1560, Loss: 0.373399\n",
      "Epoch: 13, Step: 1570, Loss: 0.390406\n",
      "Epoch: 13, Step: 1580, Loss: 0.370512\n",
      "Epoch: 13, Step: 1590, Loss: 0.35364\n",
      "Epoch: 13, Step: 1600, Loss: 0.359591\n",
      "Epoch: 13, Step: 1610, Loss: 0.364357\n",
      "Epoch: 13, Step: 1620, Loss: 0.358646\n",
      "Epoch: 13, Step: 1630, Loss: 0.374361\n",
      "Epoch: 13, Step: 1640, Loss: 0.350371\n",
      "Epoch: 13, Step: 1650, Loss: 0.37071\n",
      "Epoch: 13, Step: 1660, Loss: 0.360224\n",
      "Epoch: 13, Step: 1670, Loss: 0.415268\n",
      "Epoch: 13, Step: 1680, Loss: 0.352176\n",
      "Epoch: 13, Step: 1690, Loss: 0.382396\n",
      "Epoch: 13, Step: 1700, Loss: 0.608042\n",
      "Epoch: 13, Step: 1710, Loss: 0.349543\n",
      "Epoch: 13, Step: 1720, Loss: 0.343609\n",
      "Epoch: 13, Step: 1730, Loss: 0.371189\n",
      "Epoch: 13, Step: 1740, Loss: 0.360001\n",
      "Epoch: 13, Step: 1750, Loss: 0.349949\n",
      "Epoch: 14, Step: 1400, Loss: 0.366824\n",
      "Epoch: 14, Step: 1410, Loss: 0.341913\n",
      "Epoch: 14, Step: 1420, Loss: 0.395891\n",
      "Epoch: 14, Step: 1430, Loss: 0.341935\n",
      "Epoch: 14, Step: 1440, Loss: 0.348411\n",
      "Epoch: 14, Step: 1450, Loss: 0.345344\n",
      "Epoch: 14, Step: 1460, Loss: 0.358351\n",
      "Epoch: 14, Step: 1470, Loss: 0.373285\n",
      "Epoch: 14, Step: 1480, Loss: 0.40515\n",
      "Epoch: 14, Step: 1490, Loss: 0.423234\n",
      "Epoch: 14, Step: 1500, Loss: 0.451872\n",
      "Epoch: 14, Step: 1510, Loss: 0.343935\n",
      "Epoch: 14, Step: 1520, Loss: 0.407566\n",
      "Epoch: 14, Step: 1530, Loss: 0.33237\n",
      "Epoch: 14, Step: 1540, Loss: 0.343745\n",
      "Epoch: 14, Step: 1550, Loss: 0.333405\n",
      "Epoch: 14, Step: 1560, Loss: 0.3305\n",
      "Epoch: 14, Step: 1570, Loss: 0.336213\n",
      "Epoch: 14, Step: 1580, Loss: 0.336307\n",
      "Epoch: 14, Step: 1590, Loss: 0.324993\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch: 14, Step: 1600, Loss: 0.340815\n",
      "Epoch: 14, Step: 1610, Loss: 0.355751\n",
      "Epoch: 14, Step: 1620, Loss: 0.338762\n",
      "Epoch: 14, Step: 1630, Loss: 0.33091\n",
      "Epoch: 14, Step: 1640, Loss: 0.327738\n",
      "Epoch: 14, Step: 1650, Loss: 0.331542\n",
      "Epoch: 14, Step: 1660, Loss: 0.327806\n",
      "Epoch: 14, Step: 1670, Loss: 0.330472\n",
      "Epoch: 14, Step: 1680, Loss: 0.350952\n",
      "Epoch: 14, Step: 1690, Loss: 0.324398\n",
      "Epoch: 14, Step: 1700, Loss: 0.313403\n",
      "Epoch: 14, Step: 1710, Loss: 0.344984\n",
      "Epoch: 14, Step: 1720, Loss: 0.313066\n",
      "Epoch: 14, Step: 1730, Loss: 0.36016\n",
      "Epoch: 14, Step: 1740, Loss: 0.331601\n",
      "Epoch: 14, Step: 1750, Loss: 0.309317\n",
      "Epoch: 14, Step: 1760, Loss: 0.3295\n",
      "Epoch: 14, Step: 1770, Loss: 0.32351\n",
      "Epoch: 14, Step: 1780, Loss: 0.308079\n",
      "Epoch: 14, Step: 1790, Loss: 0.335259\n",
      "Epoch: 14, Step: 1800, Loss: 0.311301\n",
      "Epoch: 14, Step: 1810, Loss: 0.313714\n",
      "Epoch: 14, Step: 1820, Loss: 0.317822\n",
      "Epoch: 14, Step: 1830, Loss: 0.333932\n",
      "Epoch: 14, Step: 1840, Loss: 0.310814\n",
      "Epoch: 14, Step: 1850, Loss: 0.363131\n",
      "Epoch: 15, Step: 1500, Loss: 0.348903\n",
      "Epoch: 15, Step: 1510, Loss: 0.300348\n",
      "Epoch: 15, Step: 1520, Loss: 0.335144\n",
      "Epoch: 15, Step: 1530, Loss: 0.331032\n",
      "Epoch: 15, Step: 1540, Loss: 0.333766\n",
      "Epoch: 15, Step: 1550, Loss: 0.301128\n",
      "Epoch: 15, Step: 1560, Loss: 0.299748\n",
      "Epoch: 15, Step: 1570, Loss: 0.310586\n",
      "Epoch: 15, Step: 1580, Loss: 0.305445\n",
      "Epoch: 15, Step: 1590, Loss: 0.299157\n",
      "Epoch: 15, Step: 1600, Loss: 0.359682\n",
      "Epoch: 15, Step: 1610, Loss: 0.299409\n",
      "Epoch: 15, Step: 1620, Loss: 0.306632\n",
      "Epoch: 15, Step: 1630, Loss: 0.290495\n",
      "Epoch: 15, Step: 1640, Loss: 0.293051\n",
      "Epoch: 15, Step: 1650, Loss: 0.305307\n",
      "Epoch: 15, Step: 1660, Loss: 0.495225\n",
      "Epoch: 15, Step: 1670, Loss: 0.304315\n",
      "Epoch: 15, Step: 1680, Loss: 0.292441\n",
      "Epoch: 15, Step: 1690, Loss: 0.320594\n",
      "Epoch: 15, Step: 1700, Loss: 0.29792\n",
      "Epoch: 15, Step: 1710, Loss: 0.294875\n",
      "Epoch: 15, Step: 1720, Loss: 0.294347\n",
      "Epoch: 15, Step: 1730, Loss: 0.286882\n",
      "Epoch: 15, Step: 1740, Loss: 0.340842\n",
      "Epoch: 15, Step: 1750, Loss: 0.292143\n",
      "Epoch: 15, Step: 1760, Loss: 0.305612\n",
      "Epoch: 15, Step: 1770, Loss: 0.285633\n",
      "Epoch: 15, Step: 1780, Loss: 0.284001\n",
      "Epoch: 15, Step: 1790, Loss: 0.283991\n",
      "Epoch: 15, Step: 1800, Loss: 0.289136\n",
      "Epoch: 15, Step: 1810, Loss: 0.297012\n",
      "Epoch: 15, Step: 1820, Loss: 0.297368\n",
      "Epoch: 15, Step: 1830, Loss: 0.293027\n",
      "Epoch: 15, Step: 1840, Loss: 0.301655\n",
      "Epoch: 15, Step: 1850, Loss: 0.364883\n",
      "Epoch: 15, Step: 1860, Loss: 0.281393\n",
      "Epoch: 15, Step: 1870, Loss: 0.282274\n",
      "Epoch: 15, Step: 1880, Loss: 0.560257\n",
      "Epoch: 15, Step: 1890, Loss: 0.27911\n",
      "Epoch: 15, Step: 1900, Loss: 0.280833\n",
      "Epoch: 15, Step: 1910, Loss: 0.304272\n",
      "Epoch: 15, Step: 1920, Loss: 0.278016\n",
      "Epoch: 15, Step: 1930, Loss: 0.279164\n",
      "Epoch: 15, Step: 1940, Loss: 0.29595\n",
      "Epoch: 15, Step: 1950, Loss: 0.279794\n",
      "Epoch: 16, Step: 1600, Loss: 0.280285\n",
      "Epoch: 16, Step: 1610, Loss: 0.334167\n",
      "Epoch: 16, Step: 1620, Loss: 0.272037\n",
      "Epoch: 16, Step: 1630, Loss: 0.272453\n",
      "Epoch: 16, Step: 1640, Loss: 0.295399\n",
      "Epoch: 16, Step: 1650, Loss: 0.27384\n",
      "Epoch: 16, Step: 1660, Loss: 0.384137\n",
      "Epoch: 16, Step: 1670, Loss: 0.271296\n",
      "Epoch: 16, Step: 1680, Loss: 0.419486\n",
      "Epoch: 16, Step: 1690, Loss: 0.266935\n",
      "Epoch: 16, Step: 1700, Loss: 0.279058\n",
      "Epoch: 16, Step: 1710, Loss: 0.342064\n",
      "Epoch: 16, Step: 1720, Loss: 0.266753\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import tensorflow.compat.v1 as tf\n",
    "tf.disable_v2_behavior()\n",
    "from tensorflow.core.protobuf import saver_pb2\n",
    "import driving_data\n",
    "import model\n",
    "\n",
    "LOGDIR = './save'\n",
    "\n",
    "sess = tf.InteractiveSession()\n",
    "\n",
    "L2NormConst = 0.001\n",
    "\n",
    "train_vars = tf.trainable_variables()\n",
    "\n",
    "loss = tf.reduce_mean(tf.square(tf.subtract(model.y_, model.y))) + tf.add_n([tf.nn.l2_loss(v) for v in train_vars]) * L2NormConst\n",
    "train_step = tf.train.AdamOptimizer(1e-4).minimize(loss)\n",
    "sess.run(tf.global_variables_initializer())\n",
    "\n",
    "# create a summary to monitor cost tensor\n",
    "tf.summary.scalar(\"loss\", loss)\n",
    "# merge all summaries into a single op\n",
    "merged_summary_op = tf.summary.merge_all()\n",
    "\n",
    "saver = tf.train.Saver(write_version = saver_pb2.SaverDef.V2)\n",
    "\n",
    "# op to write logs to Tensorboard\n",
    "logs_path = './logs'\n",
    "summary_writer = tf.summary.FileWriter(logs_path, graph=tf.get_default_graph())\n",
    "\n",
    "epochs = 30\n",
    "batch_size = 100\n",
    "\n",
    "# train over the dataset about 30 times\n",
    "for epoch in range(epochs):\n",
    "    for i in range(int(driving_data.num_images/batch_size)):\n",
    "        xs, ys = driving_data.LoadTrainBatch(batch_size)\n",
    "        train_step.run(feed_dict={model.x: xs, model.y_: ys, model.keep_prob: 0.8})\n",
    "        if i % 10 == 0:\n",
    "            xs, ys = driving_data.LoadValBatch(batch_size)\n",
    "            loss_value = loss.eval(feed_dict={model.x:xs, model.y_: ys, model.keep_prob: 1.0})\n",
    "            print(\"Epoch: %d, Step: %d, Loss: %g\" % (epoch, epoch * batch_size + i, loss_value))\n",
    "\n",
    "    # write logs at every iteration\n",
    "    summary = merged_summary_op.eval(feed_dict={model.x:xs, model.y_: ys, model.keep_prob: 1.0})\n",
    "    summary_writer.add_summary(summary, epoch * driving_data.num_images/batch_size + i)\n",
    "\n",
    "    if i % batch_size == 0:\n",
    "        if not os.path.exists(LOGDIR):\n",
    "            os.makedirs(LOGDIR)\n",
    "            checkpoint_path = os.path.join(LOGDIR, \"model.ckpt\")\n",
    "            filename = saver.save(sess, checkpoint_path)\n",
    "            print(\"Model saved in file: %s\" % filename)\n",
    "\n",
    "print(\"Run the command line:\\n\" \\\n",
    "      \"--> tensorboard --logdir=./logs \" \\\n",
    "      \"\\nThen open http://0.0.0.0:6006/ into your web browser\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e11c0d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from itertools import islice\n",
    "from scipy import pi\n",
    "import numpy as np\n",
    "\n",
    "DATA_FOLDER = './driving_dataset/'\n",
    "TRAIN_FILE = os.path.join(DATA_FOLDER,'data.txt')\n",
    "LIMIT=None\n",
    "\n",
    "split=0.8\n",
    "x=[]\n",
    "y=[]\n",
    "with open(TRAIN_FILE) as fp:\n",
    "    for line in islice(fp,LIMIT):\n",
    "        path,angle=line.strip().split()\n",
    "        full_path = os.path.join(DATA_FOLDER,path)\n",
    "        x.append(full_path)\n",
    "    \n",
    "        y.append(float(angle)*pi/100)\n",
    "    \n",
    "y=np.array(y)\n",
    "print(\"successful\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8ec13fbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "split_index = int(len(y)*0.8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3ff52e47",
   "metadata": {},
   "outputs": [],
   "source": [
    "train_y = y[:split_index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "206a4fc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "test_y = y[split_index :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7f2d5522",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([  26.,   12.,   23.,   23.,   16.,   25.,    7.,    4.,    7.,\n",
       "           3.,    3.,    5.,    4.,    4.,    4.,    5.,    5.,   17.,\n",
       "          18.,   21.,   67.,   41.,   17.,   19.,   15.,   15.,   16.,\n",
       "          24.,   25.,  152.,  280.,  126.,  325.,  759., 2609.,  944.,\n",
       "        1806.,  727.,  245.,  258.,  132.,   30.,   20.,   53.,   75.,\n",
       "          46.,    6.,    4.,    5.,    9.]),\n",
       " array([-5.02434913, -4.87773728, -4.73112544, -4.58451359, -4.43790175,\n",
       "        -4.2912899 , -4.14467806, -3.99806621, -3.85145436, -3.70484252,\n",
       "        -3.55823067, -3.41161883, -3.26500698, -3.11839513, -2.97178329,\n",
       "        -2.82517144, -2.6785596 , -2.53194775, -2.3853359 , -2.23872406,\n",
       "        -2.09211221, -1.94550037, -1.79888852, -1.65227667, -1.50566483,\n",
       "        -1.35905298, -1.21244114, -1.06582929, -0.91921744, -0.7726056 ,\n",
       "        -0.62599375, -0.47938191, -0.33277006, -0.18615821, -0.03954637,\n",
       "         0.10706548,  0.25367732,  0.40028917,  0.54690102,  0.69351286,\n",
       "         0.84012471,  0.98673655,  1.1333484 ,  1.27996025,  1.42657209,\n",
       "         1.57318394,  1.71979578,  1.86640763,  2.01301948,  2.15963132,\n",
       "         2.30624317]),\n",
       " [<matplotlib.patches.Polygon at 0x1b85750b1c0>])"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjoAAAGdCAYAAAAbudkLAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAy5ElEQVR4nO3df1Rc9YH//xcEmAAHrvwojLMNabrFFIXaBHsI0dZ0jRg3hFq3jS7JrD3NEntskqKgNdta034raGyju7JqbHtWu/6gf1iy/eHS0F0by4b86KS0kh9at2wgFEKiwyXkB1Dy/v5Bcz9OSAgkQyZz83yccw+Ze19z532ZTu/L99xhYowxRgAAAC4UG+kBAAAATBWKDgAAcC2KDgAAcC2KDgAAcC2KDgAAcC2KDgAAcC2KDgAAcC2KDgAAcK24SA8gkk6ePKk//elPSklJUUxMTKSHAwAAJsAYoyNHjsjn8yk2dvw5m8u66PzpT3/SjBkzIj0MAABwHjo7O/XBD35w3MxlXXRSUlIkjf6iUlNTIzwaAAAwEf39/ZoxY4ZzHh/PZV10Tr1dlZqaStEBACDKTOSyEy5GBgAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArkXRAQAArnVZf3s5Ll8dHdLhw+NnMjOlnJyLMx4AwNSg6OCy09Eh5eVJx46Nn0tKkvbupewAQDSj6OCyc/jwaMl58cXRwnMme/dKy5ePZik6ABC9KDq4bOXlSXPnRnoUAICpxMXIAADAtSg6AADAtSg6AADAtSg6AADAtSg6AADAtSg6AADAtSZddN544w0tWbJEPp9PMTEx2rRp05jM3r17VVZWJsuylJKSonnz5qmjo8PZPjg4qNWrVyszM1PJyckqKyvTgQMHQvYRDAbl9/tlWZYsy5Lf71dfX19IpqOjQ0uWLFFycrIyMzO1Zs0aDQ0NTfaQAACAS0266Bw9elTXXnut6urqzrj9f//3f3XDDTfoox/9qH71q1/pd7/7nR566CFNnz7dyVRWVqqhoUH19fVqbm7WwMCASktLNTIy4mTKy8vV2tqqxsZGNTY2qrW1VX6/39k+MjKixYsX6+jRo2publZ9fb1effVVVVVVTfaQAACAW5kLIMk0NDSErLvjjjvM8uXLz3qfvr4+Ex8fb+rr6511XV1dJjY21jQ2NhpjjNmzZ4+RZLZt2+ZkWlpajCSzb98+Y4wxr732momNjTVdXV1O5pVXXjEej8fYtj2h8du2bSRNOA93CASMkUZ/XkgGABAZkzl/h/UanZMnT+rnP/+5rrrqKt1yyy3KyspSUVFRyNtbgUBAw8PDKikpcdb5fD7l5+dr69atkqSWlhZZlqWioiInM2/ePFmWFZLJz8+Xz+dzMrfccosGBwcVCATOOL7BwUH19/eHLAAAwL3CWnR6e3s1MDCgRx99VIsWLdLmzZv12c9+Vrfffru2bNkiSerp6VFCQoLS0tJC7pudna2enh4nk5WVNWb/WVlZIZns7OyQ7WlpaUpISHAyp6utrXWu+bEsSzNmzLjgYwYAAJeusM/oSNJnPvMZ3Xvvvfr4xz+uBx98UKWlpXr22WfHva8xRjExMc7t9//7QjLvt3btWtm27SydnZ0TOi4AABCdwlp0MjMzFRcXp6uvvjpkfV5envOpK6/Xq6GhIQWDwZBMb2+vM0Pj9Xp18ODBMfs/dOhQSOb0mZtgMKjh4eExMz2neDwepaamhiwAAMC9wlp0EhIS9IlPfEJvvfVWyPq3335bM2fOlCQVFhYqPj5eTU1Nzvbu7m61tbVp/vz5kqTi4mLZtq0dO3Y4me3bt8u27ZBMW1uburu7nczmzZvl8XhUWFgYzsMCAABRKm6ydxgYGNA777zj3G5vb1dra6vS09OVk5Oj+++/X3fccYc+9alP6dOf/rQaGxv105/+VL/61a8kSZZlacWKFaqqqlJGRobS09NVXV2tgoICLVy4UNLoDNCiRYtUUVGhjRs3SpJWrlyp0tJSzZ49W5JUUlKiq6++Wn6/X48//rjee+89VVdXq6KigpkaAAAwarIf6Xr99deNpDHLXXfd5WR+8IMfmI985CNm+vTp5tprrzWbNm0K2cfx48fNqlWrTHp6uklMTDSlpaWmo6MjJPPuu++aZcuWmZSUFJOSkmKWLVtmgsFgSGb//v1m8eLFJjEx0aSnp5tVq1aZEydOTPhY+Hj55YmPlwNAdJvM+TvGGGMi2LMiqr+/X5ZlybZtZoEuI7t2SYWFUiAgzZ17/hkAQGRM5vzNd10BAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXougAAADXmnTReeONN7RkyRL5fD7FxMRo06ZNZ83efffdiomJ0ZNPPhmyfnBwUKtXr1ZmZqaSk5NVVlamAwcOhGSCwaD8fr8sy5JlWfL7/err6wvJdHR0aMmSJUpOTlZmZqbWrFmjoaGhyR4SAABwqUkXnaNHj+raa69VXV3duLlNmzZp+/bt8vl8Y7ZVVlaqoaFB9fX1am5u1sDAgEpLSzUyMuJkysvL1draqsbGRjU2Nqq1tVV+v9/ZPjIyosWLF+vo0aNqbm5WfX29Xn31VVVVVU32kAAAgFuZCyDJNDQ0jFl/4MAB81d/9Vemra3NzJw50zzxxBPOtr6+PhMfH2/q6+uddV1dXSY2NtY0NjYaY4zZs2ePkWS2bdvmZFpaWowks2/fPmOMMa+99pqJjY01XV1dTuaVV14xHo/H2LY9ofHbtm0kTTgPdwgEjJFGf15IBgAQGZM5f4f9Gp2TJ0/K7/fr/vvv1zXXXDNmeyAQ0PDwsEpKSpx1Pp9P+fn52rp1qySppaVFlmWpqKjIycybN0+WZYVk8vPzQ2aMbrnlFg0ODioQCJxxbIODg+rv7w9ZAACAe4W96Dz22GOKi4vTmjVrzri9p6dHCQkJSktLC1mfnZ2tnp4eJ5OVlTXmvllZWSGZ7OzskO1paWlKSEhwMqerra11rvmxLEszZsyY9PEBAIDoEdaiEwgE9M///M96/vnnFRMTM6n7GmNC7nOm+59P5v3Wrl0r27adpbOzc1JjBAAA0SWsRefXv/61ent7lZOTo7i4OMXFxWn//v2qqqrShz70IUmS1+vV0NCQgsFgyH17e3udGRqv16uDBw+O2f+hQ4dCMqfP3ASDQQ0PD4+Z6TnF4/EoNTU1ZAEAAO4V1qLj9/v1+9//Xq2trc7i8/l0//336xe/+IUkqbCwUPHx8WpqanLu193drba2Ns2fP1+SVFxcLNu2tWPHDiezfft22bYdkmlra1N3d7eT2bx5szwejwoLC8N5WAAAIErFTfYOAwMDeuedd5zb7e3tam1tVXp6unJycpSRkRGSj4+Pl9fr1ezZsyVJlmVpxYoVqqqqUkZGhtLT01VdXa2CggItXLhQkpSXl6dFixapoqJCGzdulCStXLlSpaWlzn5KSkp09dVXy+/36/HHH9d7772n6upqVVRUMFMDAAAknceMzm9+8xvNmTNHc+bMkSTdd999mjNnjr7xjW9MeB9PPPGEbrvtNi1dulTXX3+9kpKS9NOf/lTTpk1zMi+99JIKCgpUUlKikpISfexjH9O///u/O9unTZumn//855o+fbquv/56LV26VLfddpu+853vTPaQAACAS8UYY0ykBxEp/f39sixLtm0zC3QZ2bVLKiyUAgFp7tzzzwAAImMy52++6woAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALgWRQcAALjWpIvOG2+8oSVLlsjn8ykmJkabNm1ytg0PD+urX/2qCgoKlJycLJ/Pp3/4h3/Qn/70p5B9DA4OavXq1crMzFRycrLKysp04MCBkEwwGJTf75dlWbIsS36/X319fSGZjo4OLVmyRMnJycrMzNSaNWs0NDQ02UMCAAAuNemic/ToUV177bWqq6sbs+3YsWPatWuXHnroIe3atUs//vGP9fbbb6usrCwkV1lZqYaGBtXX16u5uVkDAwMqLS3VyMiIkykvL1dra6saGxvV2Nio1tZW+f1+Z/vIyIgWL16so0ePqrm5WfX19Xr11VdVVVU12UMCAABuZS6AJNPQ0DBuZseOHUaS2b9/vzHGmL6+PhMfH2/q6+udTFdXl4mNjTWNjY3GGGP27NljJJlt27Y5mZaWFiPJ7Nu3zxhjzGuvvWZiY2NNV1eXk3nllVeMx+Mxtm1PaPy2bRtJE87DHQIBY6TRnxeSAQBExmTO31N+jY5t24qJidEVV1whSQoEAhoeHlZJSYmT8fl8ys/P19atWyVJLS0tsixLRUVFTmbevHmyLCskk5+fL5/P52RuueUWDQ4OKhAInHEsg4OD6u/vD1kAAIB7TWnROXHihB588EGVl5crNTVVktTT06OEhASlpaWFZLOzs9XT0+NksrKyxuwvKysrJJOdnR2yPS0tTQkJCU7mdLW1tc41P5ZlacaMGRd8jAAA4NI1ZUVneHhYd955p06ePKmnn376nHljjGJiYpzb7//3hWTeb+3atbJt21k6OzsncigAACBKTUnRGR4e1tKlS9Xe3q6mpiZnNkeSvF6vhoaGFAwGQ+7T29vrzNB4vV4dPHhwzH4PHToUkjl95iYYDGp4eHjMTM8pHo9HqampIQsAAHCvsBedUyXnD3/4g375y18qIyMjZHthYaHi4+PV1NTkrOvu7lZbW5vmz58vSSouLpZt29qxY4eT2b59u2zbDsm0tbWpu7vbyWzevFkej0eFhYXhPiwAABCF4iZ7h4GBAb3zzjvO7fb2drW2tio9PV0+n0+f+9zntGvXLv3sZz/TyMiIM+uSnp6uhIQEWZalFStWqKqqShkZGUpPT1d1dbUKCgq0cOFCSVJeXp4WLVqkiooKbdy4UZK0cuVKlZaWavbs2ZKkkpISXX311fL7/Xr88cf13nvvqbq6WhUVFczUAACAUZP9SNfrr79uJI1Z7rrrLtPe3n7GbZLM66+/7uzj+PHjZtWqVSY9Pd0kJiaa0tJS09HREfI47777rlm2bJlJSUkxKSkpZtmyZSYYDIZk9u/fbxYvXmwSExNNenq6WbVqlTlx4sSEj4WPl1+e+Hg5AES3yZy/Y4wxJiIN6xLQ398vy7Jk2zazQJeRXbukwkIpEJDmzj3/DAAgMiZz/ua7rgAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtRdAAAgGtNuui88cYbWrJkiXw+n2JiYrRp06aQ7cYYrVu3Tj6fT4mJiVqwYIF2794dkhkcHNTq1auVmZmp5ORklZWV6cCBAyGZYDAov98vy7JkWZb8fr/6+vpCMh0dHVqyZImSk5OVmZmpNWvWaGhoaLKHBAAAXGrSRefo0aO69tprVVdXd8bt69ev14YNG1RXV6edO3fK6/Xq5ptv1pEjR5xMZWWlGhoaVF9fr+bmZg0MDKi0tFQjIyNOpry8XK2trWpsbFRjY6NaW1vl9/ud7SMjI1q8eLGOHj2q5uZm1dfX69VXX1VVVdVkDwkAALiVuQCSTENDg3P75MmTxuv1mkcffdRZd+LECWNZlnn22WeNMcb09fWZ+Ph4U19f72S6urpMbGysaWxsNMYYs2fPHiPJbNu2zcm0tLQYSWbfvn3GGGNee+01Exsba7q6upzMK6+8Yjwej7Fte0Ljt23bSJpwHu4QCBgjjf68kAwAIDImc/4O6zU67e3t6unpUUlJibPO4/Hoxhtv1NatWyVJgUBAw8PDIRmfz6f8/Hwn09LSIsuyVFRU5GTmzZsny7JCMvn5+fL5fE7mlltu0eDgoAKBwBnHNzg4qP7+/pAFAAC4V1iLTk9PjyQpOzs7ZH12drazraenRwkJCUpLSxs3k5WVNWb/WVlZIZnTHyctLU0JCQlO5nS1tbXONT+WZWnGjBnncZQAACBaTMmnrmJiYkJuG2PGrDvd6Zkz5c8n835r166VbdvO0tnZOe6YAABAdAtr0fF6vZI0Zkalt7fXmX3xer0aGhpSMBgcN3Pw4MEx+z906FBI5vTHCQaDGh4eHjPTc4rH41FqamrIAgAA3CusRWfWrFnyer1qampy1g0NDWnLli2aP3++JKmwsFDx8fEhme7ubrW1tTmZ4uJi2batHTt2OJnt27fLtu2QTFtbm7q7u53M5s2b5fF4VFhYGM7DAgAAUSpusncYGBjQO++849xub29Xa2ur0tPTlZOTo8rKStXU1Cg3N1e5ubmqqalRUlKSysvLJUmWZWnFihWqqqpSRkaG0tPTVV1drYKCAi1cuFCSlJeXp0WLFqmiokIbN26UJK1cuVKlpaWaPXu2JKmkpERXX321/H6/Hn/8cb333nuqrq5WRUUFMzUAAGDUZD/S9frrrxtJY5a77rrLGDP6EfOHH37YeL1e4/F4zKc+9Snz5ptvhuzj+PHjZtWqVSY9Pd0kJiaa0tJS09HREZJ59913zbJly0xKSopJSUkxy5YtM8FgMCSzf/9+s3jxYpOYmGjS09PNqlWrzIkTJyZ8LHy8/PLEx8sBILpN5vwdY4wxEexZEdXf3y/LsmTbNrNAl5Fdu6TCQikQkObOPf8MACAyJnP+5ruuAACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa4W96Pz5z3/W17/+dc2aNUuJiYn68Ic/rG9961s6efKkkzHGaN26dfL5fEpMTNSCBQu0e/fukP0MDg5q9erVyszMVHJyssrKynTgwIGQTDAYlN/vl2VZsixLfr9ffX194T4kAAAQpcJedB577DE9++yzqqur0969e7V+/Xo9/vjjeuqpp5zM+vXrtWHDBtXV1Wnnzp3yer26+eabdeTIESdTWVmphoYG1dfXq7m5WQMDAyotLdXIyIiTKS8vV2trqxobG9XY2KjW1lb5/f5wHxIAAIhWJswWL15svvjFL4asu/32283y5cuNMcacPHnSeL1e8+ijjzrbT5w4YSzLMs8++6wxxpi+vj4THx9v6uvrnUxXV5eJjY01jY2Nxhhj9uzZYySZbdu2OZmWlhYjyezbt29CY7Vt20gytm2f38EiKgUCxkijPy8kAwCIjMmcv8M+o3PDDTfov/7rv/T2229Lkn73u9+publZf/u3fytJam9vV09Pj0pKSpz7eDwe3Xjjjdq6daskKRAIaHh4OCTj8/mUn5/vZFpaWmRZloqKipzMvHnzZFmWkwEAAJe3uHDv8Ktf/aps29ZHP/pRTZs2TSMjI3rkkUf093//95Kknp4eSVJ2dnbI/bKzs7V//34nk5CQoLS0tDGZU/fv6elRVlbWmMfPyspyMqcbHBzU4OCgc7u/v/88jxIAAESDsM/o/OhHP9KLL76ol19+Wbt27dILL7yg73znO3rhhRdCcjExMSG3jTFj1p3u9MyZ8uPtp7a21rlw2bIszZgxY6KHBQAAolDYi87999+vBx98UHfeeacKCgrk9/t17733qra2VpLk9XolacysS29vrzPL4/V6NTQ0pGAwOG7m4MGDYx7/0KFDY2aLTlm7dq1s23aWzs7OCztYAABwSQt70Tl27JhiY0N3O23aNOfj5bNmzZLX61VTU5OzfWhoSFu2bNH8+fMlSYWFhYqPjw/JdHd3q62tzckUFxfLtm3t2LHDyWzfvl22bTuZ03k8HqWmpoYsAADAvcJ+jc6SJUv0yCOPKCcnR9dcc41++9vfasOGDfriF78oafTtpsrKStXU1Cg3N1e5ubmqqalRUlKSysvLJUmWZWnFihWqqqpSRkaG0tPTVV1drYKCAi1cuFCSlJeXp0WLFqmiokIbN26UJK1cuVKlpaWaPXt2uA8LAABEobAXnaeeekoPPfSQ7rnnHvX29srn8+nuu+/WN77xDSfzwAMP6Pjx47rnnnsUDAZVVFSkzZs3KyUlxck88cQTiouL09KlS3X8+HHddNNNev755zVt2jQn89JLL2nNmjXOp7PKyspUV1cX7kMCAABRKsYYYyI9iEjp7++XZVmybZu3sS4ju3ZJhYVSICDNnXv+GQBAZEzm/M13XQEAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANei6AAAANeKi/QAgEvZ3r1n35aZKeXkXLyxAAAmj6IDnEFmppSUJC1ffvZMUtJoEaLsAMCli6IDnEFOzmiJOXz4zNv37h0tQYcPU3QA4FJG0QHOIieHEgMA0Y6LkQEAgGtRdAAAgGtRdAAAgGtRdAAAgGtNSdHp6urS8uXLlZGRoaSkJH384x9XIBBwthtjtG7dOvl8PiUmJmrBggXavXt3yD4GBwe1evVqZWZmKjk5WWVlZTpw4EBIJhgMyu/3y7IsWZYlv9+vvr6+qTgkAAAQhcJedILBoK6//nrFx8frP//zP7Vnzx5997vf1RVXXOFk1q9frw0bNqiurk47d+6U1+vVzTffrCNHjjiZyspKNTQ0qL6+Xs3NzRoYGFBpaalGRkacTHl5uVpbW9XY2KjGxka1trbK7/eH+5AAAEC0MmH21a9+1dxwww1n3X7y5Enj9XrNo48+6qw7ceKEsSzLPPvss8YYY/r6+kx8fLypr693Ml1dXSY2NtY0NjYaY4zZs2ePkWS2bdvmZFpaWowks2/fvgmN1bZtI8nYtj2pY0R0CwSMkUZ/RnIfAIDzM5nzd9hndH7yk5/ouuuu0+c//3llZWVpzpw5+t73vudsb29vV09Pj0pKSpx1Ho9HN954o7Zu3SpJCgQCGh4eDsn4fD7l5+c7mZaWFlmWpaKiIiczb948WZblZE43ODio/v7+kAUAALhX2IvOH//4Rz3zzDPKzc3VL37xC33pS1/SmjVr9MMf/lCS1NPTI0nKzs4OuV92drazraenRwkJCUpLSxs3k5WVNebxs7KynMzpamtrnet5LMvSjBkzLuxgAQDAJS3sRefkyZOaO3euampqNGfOHN19992qqKjQM888E5KLiYkJuW2MGbPudKdnzpQfbz9r166VbdvO0tnZOdHDAgAAUSjsRefKK6/U1VdfHbIuLy9PHR0dkiSv1ytJY2Zdent7nVker9eroaEhBYPBcTMHDx4c8/iHDh0aM1t0isfjUWpqasgCAADcK+xF5/rrr9dbb70Vsu7tt9/WzJkzJUmzZs2S1+tVU1OTs31oaEhbtmzR/PnzJUmFhYWKj48PyXR3d6utrc3JFBcXy7Zt7dixw8ls375dtm07GQAAcHkL+5d63nvvvZo/f75qamq0dOlS7dixQ88995yee+45SaNvN1VWVqqmpka5ubnKzc1VTU2NkpKSVF5eLkmyLEsrVqxQVVWVMjIylJ6erurqahUUFGjhwoWSRmeJFi1apIqKCm3cuFGStHLlSpWWlmr27NnhPiwAABCFwl50PvGJT6ihoUFr167Vt771Lc2aNUtPPvmkli1b5mQeeOABHT9+XPfcc4+CwaCKioq0efNmpaSkOJknnnhCcXFxWrp0qY4fP66bbrpJzz//vKZNm+ZkXnrpJa1Zs8b5dFZZWZnq6urCfUgAACBKxRhjTKQHESn9/f2yLEu2bXO9zmVk1y6psFAKBKS5cyO3DwDA+ZnM+ZvvugIAAK5F0QEAAK5F0QEAAK5F0QEAAK4V9k9dAZHW0SEdPnz27Xv3XryxAAAii6IDV+nokPLypGPHxs8lJUmZmRdnTACAyKHowFUOHx4tOS++OFp4ziYzU8rJuXjjAgBEBkUHrpSXx9+3AQBwMTIAAHAxig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHAtig4AAHCtKS86tbW1iomJUWVlpbPOGKN169bJ5/MpMTFRCxYs0O7du0PuNzg4qNWrVyszM1PJyckqKyvTgQMHQjLBYFB+v1+WZcmyLPn9fvX19U31IQEAgCgxpUVn586deu655/Sxj30sZP369eu1YcMG1dXVaefOnfJ6vbr55pt15MgRJ1NZWamGhgbV19erublZAwMDKi0t1cjIiJMpLy9Xa2urGhsb1djYqNbWVvn9/qk8JAAAEEWmrOgMDAxo2bJl+t73vqe0tDRnvTFGTz75pL72ta/p9ttvV35+vl544QUdO3ZML7/8siTJtm394Ac/0He/+10tXLhQc+bM0Ysvvqg333xTv/zlLyVJe/fuVWNjo77//e+ruLhYxcXF+t73vqef/exneuutt6bqsAAAQBSZsqLz5S9/WYsXL9bChQtD1re3t6unp0clJSXOOo/HoxtvvFFbt26VJAUCAQ0PD4dkfD6f8vPznUxLS4ssy1JRUZGTmTdvnizLcjKnGxwcVH9/f8gCAADcK24qdlpfX69AIKDf/OY3Y7b19PRIkrKzs0PWZ2dna//+/U4mISEhZCboVObU/Xt6epSVlTVm/1lZWU7mdLW1tfrmN785+QMCAABRKewzOp2dnfrKV76il156SdOnTz9rLiYmJuS2MWbMutOdnjlTfrz9rF27VrZtO0tnZ+e4jwcAAKJb2ItOIBBQb2+vCgsLFRcXp7i4OG3ZskX/8i//ori4OGcm5/RZl97eXmeb1+vV0NCQgsHguJmDBw+OefxDhw6NmS06xePxKDU1NWQBAADuFfaic9NNN+nNN99Ua2urs1x33XVatmyZWltb9eEPf1her1dNTU3OfYaGhrRlyxbNnz9fklRYWKj4+PiQTHd3t9ra2pxMcXGxbNvWjh07nMz27dtl27aTAQAAl7ewX6OTkpKi/Pz8kHXJycnKyMhw1ldWVqqmpka5ubnKzc1VTU2NkpKSVF5eLkmyLEsrVqxQVVWVMjIylJ6erurqahUUFDgXN+fl5WnRokWqqKjQxo0bJUkrV65UaWmpZs+eHe7DAgAAUWhKLkY+lwceeEDHjx/XPffco2AwqKKiIm3evFkpKSlO5oknnlBcXJyWLl2q48eP66abbtLzzz+vadOmOZmXXnpJa9ascT6dVVZWprq6uot+PAAA4NIUY4wxkR5EpPT398uyLNm2zfU6LrFrl1RYKAUC0ty50f84AICxJnP+5ruuAACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa1F0AACAa8VFegDAJaOjQzp8ePTfmZlSTk5kxwMAuGAUHUAaLTl5edKxY6O3k5KkvXvPWXb27h1/t/QlAIgsig4gjc7kHDsmvfji6O3ly0fXnaWlZGaOdqHly8ff7QT7EgBgilB0gPfLy5tQLCdntMCceqfrTPbuPWdfAgBMMYoOcJ5ycigwAHCp41NXAADAtSg6AADAtSg6AADAtSg6AADAtSg6AADAtcJedGpra/WJT3xCKSkpysrK0m233aa33norJGOM0bp16+Tz+ZSYmKgFCxZo9+7dIZnBwUGtXr1amZmZSk5OVllZmQ4cOBCSCQaD8vv9sixLlmXJ7/err68v3IcEAACiVNiLzpYtW/TlL39Z27ZtU1NTk/785z+rpKRER48edTLr16/Xhg0bVFdXp507d8rr9ermm2/WkSNHnExlZaUaGhpUX1+v5uZmDQwMqLS0VCMjI06mvLxcra2tamxsVGNjo1pbW+X3+8N9SAAAIFqZKdbb22skmS1bthhjjDl58qTxer3m0UcfdTInTpwwlmWZZ5991hhjTF9fn4mPjzf19fVOpqury8TGxprGxkZjjDF79uwxksy2bducTEtLi5Fk9u3bN6Gx2bZtJBnbti/4OHFpCASMkUZ/nvcdz3snYRoLAGBckzl/T/k1OrZtS5LS09MlSe3t7erp6VFJSYmT8Xg8uvHGG7V161ZJUiAQ0PDwcEjG5/MpPz/fybS0tMiyLBUVFTmZefPmybIsJ3O6wcFB9ff3hywAAMC9prToGGN033336YYbblB+fr4kqaenR5KUnZ0dks3Ozna29fT0KCEhQWlpaeNmsrKyxjxmVlaWkzldbW2tcz2PZVmaMWPGhR0gAAC4pE1p0Vm1apV+//vf65VXXhmzLSYmJuS2MWbMutOdnjlTfrz9rF27VrZtO0tnZ+dEDgMAAESpKSs6q1ev1k9+8hO9/vrr+uAHP+is93q9kjRm1qW3t9eZ5fF6vRoaGlIwGBw3c/DgwTGPe+jQoTGzRad4PB6lpqaGLAAAwL3CXnSMMVq1apV+/OMf67//+781a9askO2zZs2S1+tVU1OTs25oaEhbtmzR/PnzJUmFhYWKj48PyXR3d6utrc3JFBcXy7Zt7dixw8ls375dtm07GSAsOjqkXbtGl46OSI8GADAJYf/28i9/+ct6+eWX9R//8R9KSUlxZm4sy1JiYqJiYmJUWVmpmpoa5ebmKjc3VzU1NUpKSlJ5ebmTXbFihaqqqpSRkaH09HRVV1eroKBACxculCTl5eVp0aJFqqio0MaNGyVJK1euVGlpqWbPnh3uw8LlqqNDysuTjh0bvZ2UJO3dy9eWA0CUCHvReeaZZyRJCxYsCFn/b//2b/rCF74gSXrggQd0/Phx3XPPPQoGgyoqKtLmzZuVkpLi5J944gnFxcVp6dKlOn78uG666SY9//zzmjZtmpN56aWXtGbNGufTWWVlZaqrqwv3IeFydvjwaMl58cXR28uXj66j6ABAVAh70THGnDMTExOjdevWad26dWfNTJ8+XU899ZSeeuqps2bS09P14qkTEDCV8vIiPQIAwHngu64AAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrUXQAAIBrxUV6AMAla+/eSI8AAHCBKDqIKh0d0uHDZ98elm6SmSklJUnLl4/eTkoaXTfeAwMALkkUHUSNjg4pL086dmz83Klect5yckYb06lik5k5uo6iAwBRh6KDqHH48GjJefHF0cJzNqd6yQXJyQnDTgAAkUbRQdTJy5Pmzg3Tzk69FzaF1+Oca9dhKWYAgDOi6ODydfp7YRN9z+tUczlHQzn9Up+zSUoa3SVlBwDCj6KDS8ZFudD4/U5/L+xcUytnukh5nIZy+qU+Z7J37+juDh+m6ADAVKDo4JJw0S40PpOJvhf2/uYywYbCpT4AEFkUHVwSLuqFxheC5gIAUYWig0tKWC80jiJcsAwAU4Oigwt2rmtrJE7UZ8MFywAwtSg6uCCTubaGE/VYXLAMAFOLooMLMpFrazhRj4/LfgBg6lB0EBZRc23N+99n40s7AcD1KDpwp/cXmlMXCJ3pfbYp+bz61OCCZQCYPIoOLprxTtRhnVw50188PnUhzOnvs11oOzjbwMPYOrhgGQDOH0UHU24yJ+qwTK68v9BI/+8CoVPC8T7buQ4qjK2DC5YB4PxFfdF5+umn9fjjj6u7u1vXXHONnnzySX3yk5+M9LBcIxxfyzCRE7UUhkmQ07+gc7y/PHihxjuoU63j17+e2FdLTPDhJrKLcMyM8RYYADeJ6qLzox/9SJWVlXr66ad1/fXXa+PGjbr11lu1Z88e5Vzm/089kb9tcy6HDkm33x6er2WY8k8Wne0LOqfywuOzHdQkvxMrHCY6azYRvAUGwE1ijDEm0oM4X0VFRZo7d66eeeYZZ11eXp5uu+021dbWnvP+/f39sixLtm0rNTV1Kocq6eL9Yb2J/m2biUhKkn78Y+kDHzh75oxjHu9gp2LKYNcuqbBw7Bd0nu16nak+i79/dmn5cikQGPt22UTb6AR/X+Eot6eGeyl9FUc4jitcJnLc/AFNYOpN5vwdtTM6Q0NDCgQCevDBB0PWl5SUaOvWrWe8z+DgoAYHB53btm1LGv2FTYmentFFUs9Bye+XTgyOf5fpHunb/590xRXn/7D790sfOSY99DVp5szz348kWVdI3kRJA+OEBqT+/3vf7cOHR8+Wx4+fOZ+YOHomDeennd56a/TnjBnSRz4y+u/+/tFf5I4d0rvvjq7LyBhdN1XP+SlXXDG6DPzlFxcI/L9/S+f+Hb3fBH9fV/xlkSR5vaPLJHk8ow93rpmhqXgKz2Qyv6aL4VzHPdHxXqzfH3ApOM//OxrXqfP2hOZqTJTq6uoyksz//M//hKx/5JFHzFVXXXXG+zz88MNGEgsLCwsLC4sLls7OznP2haid0TklJiYm5LYxZsy6U9auXav77rvPuX3y5Em99957ysjIOOt9zld/f79mzJihzs7Oi/K2GC4cz1l04fmKPjxn0eVSfr6MMTpy5Ih8Pt85s1FbdDIzMzVt2jT1/OWtoVN6e3uVnZ19xvt4PB55PJ6QdVdcyHtEE5CamnrJ/Q8E4+M5iy48X9GH5yy6XKrPl2VZE8rFTvE4pkxCQoIKCwvV1NQUsr6pqUnz58+P0KgAAMClJGpndCTpvvvuk9/v13XXXafi4mI999xz6ujo0Je+9KVIDw0AAFwCorro3HHHHXr33Xf1rW99S93d3crPz9drr72mmRf6UaMw8Hg8evjhh8e8VYZLF89ZdOH5ij48Z9HFLc9XVP8dHQAAgPFE7TU6AAAA50LRAQAArkXRAQAArkXRAQAArkXRuQg+9KEPKSYmJmQ5/Tu6EFlPP/20Zs2apenTp6uwsFC//vWvIz0knMW6devGvJ684f4iHZy3N954Q0uWLJHP51NMTIw2bdoUst0Yo3Xr1snn8ykxMVELFizQ7t27IzNYSDr3c/aFL3xhzGtu3rx5kRnseaDoXCSnPgJ/avn6178e6SHhL370ox+psrJSX/va1/Tb3/5Wn/zkJ3Xrrbeqo6Mj0kPDWVxzzTUhr6c333wz0kPCXxw9elTXXnut6urqzrh9/fr12rBhg+rq6rRz5055vV7dfPPNOnLkyEUeKU4513MmSYsWLQp5zb322msXcYQXJqr/jk40SUlJ4b86L1EbNmzQihUr9I//+I+SpCeffFK/+MUv9Mwzz6i2tjbCo8OZxMXF8Xq6RN1666269dZbz7jNGKMnn3xSX/va13T77bdLkl544QVlZ2fr5Zdf1t13330xh4q/GO85O8Xj8UTta44ZnYvkscceU0ZGhj7+8Y/rkUce0dDQUKSHBElDQ0MKBAIqKSkJWV9SUqKtW7dGaFQ4lz/84Q/y+XyaNWuW7rzzTv3xj3+M9JAwAe3t7erp6Ql5vXk8Ht1444283i5xv/rVr5SVlaWrrrpKFRUV6u3tjfSQJowZnYvgK1/5iubOnau0tDTt2LFDa9euVXt7u77//e9HemiXvcOHD2tkZGTMF8FmZ2eP+cJYXBqKior0wx/+UFdddZUOHjyob3/725o/f752796tjIyMSA8P4zj1mjrT623//v2RGBIm4NZbb9XnP/95zZw5U+3t7XrooYf0N3/zNwoEAlHxV5MpOudp3bp1+uY3vzluZufOnbruuut07733Ous+9rGPKS0tTZ/73OecWR5EXkxMTMhtY8yYdbg0vH+KvaCgQMXFxfrrv/5rvfDCC7rvvvsiODJMFK+36HLHHXc4/87Pz9d1112nmTNn6uc//7nzFuSljKJznlatWqU777xz3MyHPvShM64/dbX6O++8Q9GJsMzMTE2bNm3M7E1vb++Y/+rEpSk5OVkFBQX6wx/+EOmh4BxOXePR09OjK6+80lnP6y26XHnllZo5c2bUvOYoOucpMzNTmZmZ53Xf3/72t5IU8kJHZCQkJKiwsFBNTU367Gc/66xvamrSZz7zmQiODBM1ODiovXv36pOf/GSkh4JzmDVrlrxer5qamjRnzhxJo9fJbdmyRY899liER4eJevfdd9XZ2Rk15zCKzhRraWnRtm3b9OlPf1qWZWnnzp269957VVZWppycnEgPD5Luu+8++f1+XXfddSouLtZzzz2njo4OfelLX4r00HAG1dXVWrJkiXJyctTb26tvf/vb6u/v11133RXpoUHSwMCA3nnnHed2e3u7WltblZ6erpycHFVWVqqmpka5ubnKzc1VTU2NkpKSVF5eHsFRX97Ge87S09O1bt06/d3f/Z2uvPJK/d///Z/+6Z/+SZmZmSH/cXhJM5hSgUDAFBUVGcuyzPTp083s2bPNww8/bI4ePRrpoeF9/vVf/9XMnDnTJCQkmLlz55otW7ZEekg4izvuuMNceeWVJj4+3vh8PnP77beb3bt3R3pY+IvXX3/dSBqz3HXXXcYYY06ePGkefvhh4/V6jcfjMZ/61KfMm2++GdlBX+bGe86OHTtmSkpKzAc+8AETHx9vcnJyzF133WU6OjoiPewJizHGmEiVLAAAgKnE39EBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACuRdEBAACu9f8DHbagGbbmT+YAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.hist(train_y,bins=50,color=\"blue\",histtype='step')\n",
    "plt.hist(test_y,bins=50,color=\"red\",histtype='step')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1659abf1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6192510707709883"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"#regression problem\n",
    "I/p = images\n",
    "O/p = angle\n",
    "MSE \"\"\"\n",
    "\n",
    "train_mean_y = np.mean(train_y)\n",
    "np.mean(np.square(test_y-train_mean_y))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ad686f7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6184869809073972"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(np.square(test_y-0.0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1411719",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b36fddbd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\compat\\v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:From C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\util\\dispatch.py:1176: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.\n"
     ]
    }
   ],
   "source": [
    "import tensorflow.compat.v1 as tf\n",
    "tf.disable_v2_behavior()\n",
    "import scipy\n",
    "\n",
    "def weight_variable(shape):\n",
    "  initial = tf.truncated_normal(shape, stddev=0.1)\n",
    "  return tf.Variable(initial)\n",
    "\n",
    "def bias_variable(shape):\n",
    "  initial = tf.constant(0.1, shape=shape)\n",
    "  return tf.Variable(initial)\n",
    "\n",
    "def conv2d(x, W, stride):\n",
    "  return tf.nn.conv2d(x, W, strides=[1, stride, stride, 1], padding='VALID')\n",
    "\n",
    "x = tf.placeholder(tf.float32, shape=[None, 66, 200, 3])\n",
    "y_ = tf.placeholder(tf.float32, shape=[None, 1])\n",
    "\n",
    "x_image = x\n",
    "\n",
    "#first convolutional layer\n",
    "W_conv1 = weight_variable([5, 5, 3, 24])\n",
    "b_conv1 = bias_variable([24])\n",
    "\n",
    "h_conv1 = tf.nn.relu(conv2d(x_image, W_conv1, 2) + b_conv1)\n",
    "\n",
    "#second convolutional layer\n",
    "W_conv2 = weight_variable([5, 5, 24, 36])\n",
    "b_conv2 = bias_variable([36])\n",
    "\n",
    "h_conv2 = tf.nn.relu(conv2d(h_conv1, W_conv2, 2) + b_conv2)\n",
    "\n",
    "#third convolutional layer\n",
    "W_conv3 = weight_variable([5, 5, 36, 48])\n",
    "b_conv3 = bias_variable([48])\n",
    "\n",
    "h_conv3 = tf.nn.relu(conv2d(h_conv2, W_conv3, 2) + b_conv3)\n",
    "\n",
    "#fourth convolutional layer\n",
    "W_conv4 = weight_variable([3, 3, 48, 64])\n",
    "b_conv4 = bias_variable([64])\n",
    "\n",
    "h_conv4 = tf.nn.relu(conv2d(h_conv3, W_conv4, 1) + b_conv4)\n",
    "\n",
    "#fifth convolutional layer\n",
    "W_conv5 = weight_variable([3, 3, 64, 64])\n",
    "b_conv5 = bias_variable([64])\n",
    "\n",
    "h_conv5 = tf.nn.relu(conv2d(h_conv4, W_conv5, 1) + b_conv5)\n",
    "\n",
    "#FCL 1\n",
    "W_fc1 = weight_variable([1152, 1164])\n",
    "b_fc1 = bias_variable([1164])\n",
    "\n",
    "h_conv5_flat = tf.reshape(h_conv5, [-1, 1152])\n",
    "h_fc1 = tf.nn.relu(tf.matmul(h_conv5_flat, W_fc1) + b_fc1)\n",
    "\n",
    "keep_prob = tf.placeholder(tf.float32)\n",
    "h_fc1_drop = tf.nn.dropout(h_fc1, keep_prob)\n",
    "\n",
    "#FCL 2\n",
    "W_fc2 = weight_variable([1164, 100])\n",
    "b_fc2 = bias_variable([100])\n",
    "\n",
    "h_fc2 = tf.nn.relu(tf.matmul(h_fc1_drop, W_fc2) + b_fc2)\n",
    "\n",
    "h_fc2_drop = tf.nn.dropout(h_fc2, keep_prob)\n",
    "\n",
    "#FCL 3\n",
    "W_fc3 = weight_variable([100, 50])\n",
    "b_fc3 = bias_variable([50])\n",
    "\n",
    "h_fc3 = tf.nn.relu(tf.matmul(h_fc2_drop, W_fc3) + b_fc3)\n",
    "\n",
    "h_fc3_drop = tf.nn.dropout(h_fc3, keep_prob)\n",
    "\n",
    "#FCL 3\n",
    "W_fc4 = weight_variable([50, 10])\n",
    "b_fc4 = bias_variable([10])\n",
    "\n",
    "h_fc4 = tf.nn.relu(tf.matmul(h_fc3_drop, W_fc4) + b_fc4)\n",
    "\n",
    "h_fc4_drop = tf.nn.dropout(h_fc4, keep_prob)\n",
    "\n",
    "#Output\n",
    "W_fc5 = weight_variable([10, 1])\n",
    "b_fc5 = bias_variable([1])\n",
    "\n",
    "y = tf.multiply(tf.atan(tf.matmul(h_fc4_drop, W_fc5) + b_fc5), 2) #scale the atan output\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "786e7d43",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'driving_dataset/data.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_19648\\2173663350.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     11\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[1;31m#read data.txt\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 13\u001b[1;33m \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"driving_dataset/data.txt\"\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     14\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     15\u001b[0m         \u001b[0mxs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"driving_dataset/\"\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'driving_dataset/data.txt'"
     ]
    }
   ],
   "source": [
    "import cv2\n",
    "import random\n",
    "import numpy as np\n",
    "\n",
    "xs = []\n",
    "ys = []\n",
    "\n",
    "#points to the end of the last batch\n",
    "train_batch_pointer = 0\n",
    "val_batch_pointer = 0\n",
    "\n",
    "#read data.txt\n",
    "with open(\"driving_dataset/data.txt\") as f:\n",
    "    for line in f:\n",
    "        xs.append(\"driving_dataset/\" + line.split()[0])\n",
    "        #the paper by Nvidia uses the inverse of the turning radius,\n",
    "        #but steering wheel angle is proportional to the inverse of turning radius\n",
    "        #so the steering wheel angle in radians is used as the output\n",
    "        ys.append(float(line.split()[1]) * 3.14159265 / 180)\n",
    "\n",
    "#get number of images\n",
    "num_images = len(xs)\n",
    "\n",
    "#shuffle list of images\n",
    "c = list(zip(xs, ys))\n",
    "random.shuffle(c)\n",
    "xs, ys = zip(*c)\n",
    "\n",
    "train_xs = xs[:int(len(xs) * 0.8)]\n",
    "train_ys = ys[:int(len(xs) * 0.8)]\n",
    "\n",
    "val_xs = xs[-int(len(xs) * 0.2):]\n",
    "val_ys = ys[-int(len(xs) * 0.2):]\n",
    "\n",
    "num_train_images = len(train_xs)\n",
    "num_val_images = len(val_xs)\n",
    "\n",
    "def LoadTrainBatch(batch_size):\n",
    "    global train_batch_pointer\n",
    "    x_out = []\n",
    "    y_out = []\n",
    "    for i in range(0, batch_size):\n",
    "        x_out.append(cv2.resize(cv2.imread(train_xs[(train_batch_pointer + i) % num_train_images])[-150:], (200, 66)) / 255.0)\n",
    "        y_out.append([train_ys[(train_batch_pointer + i) % num_train_images]])\n",
    "    train_batch_pointer += batch_size\n",
    "    return x_out, y_out\n",
    "\n",
    "def LoadValBatch(batch_size):\n",
    "    global val_batch_pointer\n",
    "    x_out = []\n",
    "    y_out = []\n",
    "    for i in range(0, batch_size):\n",
    "        x_out.append(cv2.resize(cv2.imread(val_xs[(val_batch_pointer + i) % num_val_images])[-150:], (200, 66)) / 255.0)\n",
    "        y_out.append([val_ys[(val_batch_pointer + i) % num_val_images]])\n",
    "    val_batch_pointer += batch_size\n",
    "    return x_out, y_out\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f00b7638",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting opencv-python\n",
      "  Downloading opencv_python-4.6.0.66-cp36-abi3-win_amd64.whl (35.6 MB)\n",
      "     ---------------------------------------- 35.6/35.6 MB 9.0 MB/s eta 0:00:00\n",
      "Requirement already satisfied: numpy>=1.19.3 in c:\\users\\manis\\anaconda3\\lib\\site-packages (from opencv-python) (1.21.5)\n",
      "Installing collected packages: opencv-python\n",
      "Successfully installed opencv-python-4.6.0.66\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install opencv-python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ac55179a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:tensorflow:Restoring parameters from save/model.ckpt\n"
     ]
    },
    {
     "ename": "NotFoundError",
     "evalue": "Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:\n\nGraph execution error:\n\nDetected at node 'save_6/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_6/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_6/RestoreV2}}]]\n\nOriginal stack trace for 'save_6/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1377\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1378\u001b[1;33m       \u001b[1;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1379\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mOpError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run_fn\u001b[1;34m(feed_dict, fetch_list, target_list, options, run_metadata)\u001b[0m\n\u001b[0;32m   1360\u001b[0m       \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_extend_graph\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1361\u001b[1;33m       return self._call_tf_sessionrun(options, feed_dict, fetch_list,\n\u001b[0m\u001b[0;32m   1362\u001b[0m                                       target_list, run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_call_tf_sessionrun\u001b[1;34m(self, options, feed_dict, fetch_list, target_list, run_metadata)\u001b[0m\n\u001b[0;32m   1453\u001b[0m                           run_metadata):\n\u001b[1;32m-> 1454\u001b[1;33m     return tf_session.TF_SessionRun_wrapper(self._session, options, feed_dict,\n\u001b[0m\u001b[0;32m   1455\u001b[0m                                             \u001b[0mfetch_list\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtarget_list\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Tensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_6/RestoreV2}}]]",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1414\u001b[0m       \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1415\u001b[1;33m         sess.run(self.saver_def.restore_op_name,\n\u001b[0m\u001b[0;32m   1416\u001b[0m                  {self.saver_def.filename_tensor_name: save_path})\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36mrun\u001b[1;34m(self, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m    967\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 968\u001b[1;33m       result = self._run(None, fetches, feed_dict, options_ptr,\n\u001b[0m\u001b[0;32m    969\u001b[0m                          run_metadata_ptr)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run\u001b[1;34m(self, handle, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1190\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mfinal_fetches\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mfinal_targets\u001b[0m \u001b[1;32mor\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mfeed_dict_tensor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1191\u001b[1;33m       results = self._do_run(handle, final_targets, final_fetches,\n\u001b[0m\u001b[0;32m   1192\u001b[0m                              feed_dict_tensor, options, run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_run\u001b[1;34m(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1370\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mhandle\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1371\u001b[1;33m       return self._do_call(_run_fn, feeds, fetches, targets, options,\n\u001b[0m\u001b[0;32m   1372\u001b[0m                            run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1396\u001b[0m                     'disable_meta_optimizer = True')\n\u001b[1;32m-> 1397\u001b[1;33m       \u001b[1;32mraise\u001b[0m \u001b[0mtype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnode_def\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mop\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# pylint: disable=no-value-for-parameter\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1398\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Graph execution error:\n\nDetected at node 'save_6/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_6/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_6/RestoreV2}}]]\n\nOriginal stack trace for 'save_6/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36mget_tensor\u001b[1;34m(self, tensor_str)\u001b[0m\n\u001b[0;32m     65\u001b[0m   \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 66\u001b[1;33m     return CheckpointReader.CheckpointReader_GetTensor(\n\u001b[0m\u001b[0;32m     67\u001b[0m         self, compat.as_bytes(tensor_str))\n",
      "\u001b[1;31mRuntimeError\u001b[0m: _CHECKPOINTABLE_OBJECT_GRAPH not found in checkpoint file",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1425\u001b[0m       \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1426\u001b[1;33m         \u001b[0mnames_to_keys\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mobject_graph_key_mapping\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msave_path\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1427\u001b[0m       \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNotFoundError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mobject_graph_key_mapping\u001b[1;34m(checkpoint_path)\u001b[0m\n\u001b[0;32m   1747\u001b[0m   \u001b[0mreader\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpy_checkpoint_reader\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNewCheckpointReader\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcheckpoint_path\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1748\u001b[1;33m   \u001b[0mobject_graph_string\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mreader\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_tensor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtrackable\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mOBJECT_GRAPH_PROTO_KEY\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1749\u001b[0m   \u001b[0mobject_graph_proto\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mtrackable_object_graph_pb2\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTrackableObjectGraph\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36mget_tensor\u001b[1;34m(self, tensor_str)\u001b[0m\n\u001b[0;32m     70\u001b[0m   \u001b[1;32mexcept\u001b[0m \u001b[0mRuntimeError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 71\u001b[1;33m     \u001b[0merror_translator\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     72\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36merror_translator\u001b[1;34m(e)\u001b[0m\n\u001b[0;32m     30\u001b[0m       'matching files for') in error_message:\n\u001b[1;32m---> 31\u001b[1;33m     \u001b[1;32mraise\u001b[0m \u001b[0merrors_impl\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNotFoundError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0merror_message\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     32\u001b[0m   elif 'Sliced checkpoints are not supported' in error_message or (\n",
      "\u001b[1;31mNotFoundError\u001b[0m: _CHECKPOINTABLE_OBJECT_GRAPH not found in checkpoint file",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[0msess\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mInteractiveSession\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[0msaver\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtrain\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSaver\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m \u001b[0msaver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrestore\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msess\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"save/model.ckpt\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m \u001b[0mimg\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcv2\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mimread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'steering_wheel_image.jpg'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1429\u001b[0m         \u001b[1;31m# is a graph mismatch. Re-raise the original error with\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1430\u001b[0m         \u001b[1;31m# a helpful message (b/110263146)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1431\u001b[1;33m         raise _wrap_restore_error_with_msg(\n\u001b[0m\u001b[0;32m   1432\u001b[0m             err, \"a Variable name or other graph key that is missing\")\n\u001b[0;32m   1433\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:\n\nGraph execution error:\n\nDetected at node 'save_6/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_6/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_6/RestoreV2}}]]\n\nOriginal stack trace for 'save_6/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2347941609.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n"
     ]
    }
   ],
   "source": [
    "import tensorflow.compat.v1 as tf\n",
    "tf.disable_v2_behavior()\n",
    "import model\n",
    "import cv2\n",
    "from subprocess import call\n",
    "import os\n",
    "\n",
    "#check if on windows OS\n",
    "windows = False\n",
    "if os.name == 'nt':\n",
    "    windows = True\n",
    "\n",
    "sess = tf.InteractiveSession()\n",
    "saver = tf.train.Saver()\n",
    "saver.restore(sess, \"save/model.ckpt\")\n",
    "\n",
    "img = cv2.imread('steering_wheel_image.jpg',0)\n",
    "rows,cols = img.shape\n",
    "\n",
    "smoothed_angle = 0\n",
    "\n",
    "cap = cv2.VideoCapture(0)\n",
    "while(cv2.waitKey(10) != ord('q')):\n",
    "    ret, frame = cap.read()\n",
    "    image = cv2.resize(frame, (200, 66)) / 255.0\n",
    "    degrees = model.y.eval(feed_dict={model.x: [image], model.keep_prob: 1.0})[0][0] * 180 / 3.14159265\n",
    "    if not windows:\n",
    "        call(\"clear\")\n",
    "    print(\"Predicted steering angle: \" + str(degrees) + \" degrees\")\n",
    "    cv2.imshow('frame', frame)\n",
    "    #make smooth angle transitions by turning the steering wheel based on the difference of the current angle\n",
    "    #and the predicted angle\n",
    "    smoothed_angle += 0.2 * pow(abs((degrees - smoothed_angle)), 2.0 / 3.0) * (degrees - smoothed_angle) / abs(degrees - smoothed_angle)\n",
    "    M = cv2.getRotationMatrix2D((cols/2,rows/2),-smoothed_angle,1)\n",
    "    dst = cv2.warpAffine(img,M,(cols,rows))\n",
    "    cv2.imshow(\"steering wheel\", dst)\n",
    "\n",
    "cap.release()\n",
    "cv2.destroyAllWindows()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5a7a1b69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:tensorflow:Restoring parameters from save/model.ckpt\n"
     ]
    },
    {
     "ename": "NotFoundError",
     "evalue": "Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:\n\nGraph execution error:\n\nDetected at node 'save_5/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_5/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_5/RestoreV2}}]]\n\nOriginal stack trace for 'save_5/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1377\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1378\u001b[1;33m       \u001b[1;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1379\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mOpError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run_fn\u001b[1;34m(feed_dict, fetch_list, target_list, options, run_metadata)\u001b[0m\n\u001b[0;32m   1360\u001b[0m       \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_extend_graph\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1361\u001b[1;33m       return self._call_tf_sessionrun(options, feed_dict, fetch_list,\n\u001b[0m\u001b[0;32m   1362\u001b[0m                                       target_list, run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_call_tf_sessionrun\u001b[1;34m(self, options, feed_dict, fetch_list, target_list, run_metadata)\u001b[0m\n\u001b[0;32m   1453\u001b[0m                           run_metadata):\n\u001b[1;32m-> 1454\u001b[1;33m     return tf_session.TF_SessionRun_wrapper(self._session, options, feed_dict,\n\u001b[0m\u001b[0;32m   1455\u001b[0m                                             \u001b[0mfetch_list\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtarget_list\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Tensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_5/RestoreV2}}]]",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1414\u001b[0m       \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1415\u001b[1;33m         sess.run(self.saver_def.restore_op_name,\n\u001b[0m\u001b[0;32m   1416\u001b[0m                  {self.saver_def.filename_tensor_name: save_path})\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36mrun\u001b[1;34m(self, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m    967\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 968\u001b[1;33m       result = self._run(None, fetches, feed_dict, options_ptr,\n\u001b[0m\u001b[0;32m    969\u001b[0m                          run_metadata_ptr)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run\u001b[1;34m(self, handle, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1190\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mfinal_fetches\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mfinal_targets\u001b[0m \u001b[1;32mor\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mfeed_dict_tensor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1191\u001b[1;33m       results = self._do_run(handle, final_targets, final_fetches,\n\u001b[0m\u001b[0;32m   1192\u001b[0m                              feed_dict_tensor, options, run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_run\u001b[1;34m(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1370\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mhandle\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1371\u001b[1;33m       return self._do_call(_run_fn, feeds, fetches, targets, options,\n\u001b[0m\u001b[0;32m   1372\u001b[0m                            run_metadata)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1396\u001b[0m                     'disable_meta_optimizer = True')\n\u001b[1;32m-> 1397\u001b[1;33m       \u001b[1;32mraise\u001b[0m \u001b[0mtype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnode_def\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mop\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# pylint: disable=no-value-for-parameter\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1398\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Graph execution error:\n\nDetected at node 'save_5/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_5/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_5/RestoreV2}}]]\n\nOriginal stack trace for 'save_5/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36mget_tensor\u001b[1;34m(self, tensor_str)\u001b[0m\n\u001b[0;32m     65\u001b[0m   \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 66\u001b[1;33m     return CheckpointReader.CheckpointReader_GetTensor(\n\u001b[0m\u001b[0;32m     67\u001b[0m         self, compat.as_bytes(tensor_str))\n",
      "\u001b[1;31mRuntimeError\u001b[0m: _CHECKPOINTABLE_OBJECT_GRAPH not found in checkpoint file",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1425\u001b[0m       \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1426\u001b[1;33m         \u001b[0mnames_to_keys\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mobject_graph_key_mapping\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msave_path\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1427\u001b[0m       \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNotFoundError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mobject_graph_key_mapping\u001b[1;34m(checkpoint_path)\u001b[0m\n\u001b[0;32m   1747\u001b[0m   \u001b[0mreader\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpy_checkpoint_reader\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNewCheckpointReader\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcheckpoint_path\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1748\u001b[1;33m   \u001b[0mobject_graph_string\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mreader\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_tensor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtrackable\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mOBJECT_GRAPH_PROTO_KEY\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1749\u001b[0m   \u001b[0mobject_graph_proto\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mtrackable_object_graph_pb2\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTrackableObjectGraph\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36mget_tensor\u001b[1;34m(self, tensor_str)\u001b[0m\n\u001b[0;32m     70\u001b[0m   \u001b[1;32mexcept\u001b[0m \u001b[0mRuntimeError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 71\u001b[1;33m     \u001b[0merror_translator\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     72\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\py_checkpoint_reader.py\u001b[0m in \u001b[0;36merror_translator\u001b[1;34m(e)\u001b[0m\n\u001b[0;32m     30\u001b[0m       'matching files for') in error_message:\n\u001b[1;32m---> 31\u001b[1;33m     \u001b[1;32mraise\u001b[0m \u001b[0merrors_impl\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mNotFoundError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0merror_message\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     32\u001b[0m   elif 'Sliced checkpoints are not supported' in error_message or (\n",
      "\u001b[1;31mNotFoundError\u001b[0m: _CHECKPOINTABLE_OBJECT_GRAPH not found in checkpoint file",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mNotFoundError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[0msess\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mInteractiveSession\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[0msaver\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtrain\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSaver\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m \u001b[0msaver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrestore\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msess\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"save/model.ckpt\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m \u001b[0mimg\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcv2\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mimread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'steering_wheel_image.jpg'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\u001b[0m in \u001b[0;36mrestore\u001b[1;34m(self, sess, save_path)\u001b[0m\n\u001b[0;32m   1429\u001b[0m         \u001b[1;31m# is a graph mismatch. Re-raise the original error with\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1430\u001b[0m         \u001b[1;31m# a helpful message (b/110263146)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1431\u001b[1;33m         raise _wrap_restore_error_with_msg(\n\u001b[0m\u001b[0;32m   1432\u001b[0m             err, \"a Variable name or other graph key that is missing\")\n\u001b[0;32m   1433\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNotFoundError\u001b[0m: Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:\n\nGraph execution error:\n\nDetected at node 'save_5/RestoreV2' defined at (most recent call last):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n      app.launch_new_instance()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n      self.io_loop.start()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n      await self.process_one()\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n      await dispatch(*args)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n      await result\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n      reply_content = await reply_content\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n      result = self._run_cell(\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n      return runner(coro)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n      coro.send(None)\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n      if (await self.run_code(code, result,  async_=asy)):\n    File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n      saver = tf.train.Saver()\nNode: 'save_5/RestoreV2'\nTensor name \"Variable_20\" not found in checkpoint files save/model.ckpt\n\t [[{{node save_5/RestoreV2}}]]\n\nOriginal stack trace for 'save_5/RestoreV2':\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 712, in start\n    self.io_loop.start()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 510, in dispatch_queue\n    await self.process_one()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 499, in process_one\n    await dispatch(*args)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 406, in dispatch_shell\n    await result\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 730, in execute_request\n    reply_content = await reply_content\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 390, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 528, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2914, in run_cell\n    result = self._run_cell(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2960, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 78, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3185, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3377, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3457, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\265683558.py\", line 14, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 931, in __init__\n    self.build()\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 943, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 971, in _build\n    self.saver_def = self._builder._build_internal(  # pylint: disable=protected-access\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 540, in _build_internal\n    restore_op = self._AddRestoreOps(filename_tensor, saveables,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 360, in _AddRestoreOps\n    all_tensors = self.bulk_restore(filename_tensor, saveables, preferred_shard,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\training\\saver.py\", line 608, in bulk_restore\n    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\ops\\gen_io_ops.py\", line 1516, in restore_v2\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\manis\\anaconda3\\lib\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n"
     ]
    }
   ],
   "source": [
    "import tensorflow.compat.v1 as tf\n",
    "tf.disable_v2_behavior()\n",
    "import model\n",
    "import cv2\n",
    "from subprocess import call\n",
    "import os\n",
    "\n",
    "#check if on windows OS\n",
    "windows = False\n",
    "if os.name == 'nt':\n",
    "    windows = True\n",
    "\n",
    "sess = tf.InteractiveSession()\n",
    "saver = tf.train.Saver()\n",
    "saver.restore(sess, \"save/model.ckpt\")\n",
    "\n",
    "img = cv2.imread('steering_wheel_image.jpg',0)\n",
    "rows,cols = img.shape\n",
    "\n",
    "smoothed_angle = 0\n",
    "\n",
    "i = 0\n",
    "while(cv2.waitKey(10) != ord('q')):\n",
    "    full_image = cv2.imread(\"driving_dataset/\" + str(i) + \".jpg\")\n",
    "    image = cv2.resize(full_image[-150:], (200, 66)) / 255.0\n",
    "    degrees = model.y.eval(feed_dict={model.x: [image], model.keep_prob: 1.0})[0][0] * 180.0 / 3.14159265\n",
    "    if not windows:\n",
    "        call(\"clear\")\n",
    "    print(\"Predicted steering angle: \" + str(degrees) + \" degrees\")\n",
    "    cv2.imshow(\"frame\", full_image)\n",
    "    #make smooth angle transitions by turning the steering wheel based on the difference of the current angle\n",
    "    #and the predicted angle\n",
    "    smoothed_angle += 0.2 * pow(abs((degrees - smoothed_angle)), 2.0 / 3.0) * (degrees - smoothed_angle) / abs(degrees - smoothed_angle)\n",
    "    M = cv2.getRotationMatrix2D((cols/2,rows/2),-smoothed_angle,1)\n",
    "    dst = cv2.warpAffine(img,M,(cols,rows))\n",
    "    cv2.imshow(\"steering wheel\", dst)\n",
    "    i += 1\n",
    "\n",
    "cv2.destroyAllWindows()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "6c1251a4",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "cannot assign to operator (2911169554.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;36m  File \u001b[1;32m\"C:\\Users\\manis\\AppData\\Local\\Temp\\ipykernel_18956\\2911169554.py\"\u001b[1;36m, line \u001b[1;32m1\u001b[0m\n\u001b[1;33m    tensorboard --logdir=./logs\u001b[0m\n\u001b[1;37m    ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m cannot assign to operator\n"
     ]
    }
   ],
   "source": [
    "tensorboard --logdir=./logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4bb62892",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "EnvironmentFileNotFound: 'C:\\Users\\manis\\Downloads\\driving_dataset\\environments-gpu.yml' file not found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda env create -f environments-gpu.yml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "b5a4c716",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "EnvironmentFileNotFound: 'C:\\Users\\manis\\Downloads\\driving_dataset\\environments.yml' file not found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda env create -f environments.yml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bbb31901",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "restore() got an unexpected keyword argument 'path'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_10116\\196866194.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mconv_layer1\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m...\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m \u001b[0msaver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrestore\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m...\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m: restore() got an unexpected keyword argument 'path'"
     ]
    }
   ],
   "source": [
    "saver = tf.train.Saver()\n",
    "conv_layer1 = ...\n",
    "\n",
    "saver.restore(path=...)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c97f367d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
